commit edcc102f57a201f94c4aee34ab5b7d5fb6b2aed5
Author: Anil Madhavapeddy <anil@recoil.org>
Date:   Sun Jun 21 22:56:47 2020 +0100

    add `-fcommon` unconditionally to fix build on gcc10

diff --git a/configure b/configure
index 4ea1498cf..b3162734c 100755
--- a/configure
+++ b/configure
@@ -444,6 +444,8 @@ case "$bytecc,$target" in
     bytecccompopts="-fno-defer-pop $gcc_warnings";;
 esac
 
+bytecccompopts="-fcommon $bytecccompopts"
+
 # Configure compiler to use in further tests
 
 cc="$bytecc -O $bytecclinkopts"
@@ -907,6 +909,7 @@ case "$arch,$nativecc,$system,$target" in
   *,gcc*,*,*)          nativecccompopts="$gcc_warnings";;
 esac
 
+nativecccompopts="-fcommon $nativecccompopts"
 asppprofflags='-DPROFILING'
 
 case "$arch,$system" in
