diff -rN -u old-kaputt.x9c.fr/src/library/enumerator.ml new-kaputt.x9c.fr/src/library/enumerator.ml
--- old-kaputt.x9c.fr/src/library/enumerator.ml	2018-01-18 17:59:10.553942797 +0100
+++ new-kaputt.x9c.fr/src/library/enumerator.ml	2018-01-18 17:59:10.553942797 +0100
@@ -135,11 +135,11 @@
   create_state_based
     (fun () -> (Array.init len (fun _ -> ch)))
     (fun s ->
-      let res = String.create len in
+      let res = Bytes.create len in
       for i = 0 to pred len do
         res.[i] <- match s.State.curr.(i) with Cons (hd, _) -> hd | _ -> assert false
       done;
-      res)
+      Bytes.to_string res)
     Utils.string_of_string
 
 let float x y steps =
diff -rN -u old-kaputt.x9c.fr/src/library/generator.ml new-kaputt.x9c.fr/src/library/generator.ml
--- old-kaputt.x9c.fr/src/library/generator.ml	2018-01-18 17:59:10.553942797 +0100
+++ new-kaputt.x9c.fr/src/library/generator.ml	2018-01-18 17:59:10.557942793 +0100
@@ -171,11 +171,11 @@
 let string (gen_l, _) (gen_c, _) =
   (fun r ->
     let len = gen_l r in
-    let res = String.create len in
+    let res = Bytes.create len in
     for i = 0 to pred len do
       res.[i] <- gen_c r
     done;
-    res),
+    Bytes.to_string res),
   Utils.string_of_string
 
 let strings sep (gen_l, _) (gen_s, _) =
