diff --git a/Makefile.in b/Makefile.in
index 6d16a15..467bfcb 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1821,7 +1821,8 @@ clean::
 
 # the generic rule cannot be applied since ocaml would confuse
 # lib/why3/why3extract.cmi for the interface (!!!)
-src/tools/why3extract.cmx: src/tools/why3extract.ml
+# same for ocaml/lablgtk2/gMain.cmi on case-insensitive filesystems
+src/tools/why3extract.cmx src/ide/gmain.cmx: %.cmx: %.ml
 	$(SHOW) 'Ocamlopt $<'
 	$(HIDE)$(OCAMLOPT) -c $(OFLAGS) $<
 
