diff -Naur lablgtk-2.16.0.orig/configure lablgtk-2.16.0/configure
--- lablgtk-2.16.0.orig/configure	2012-08-23 12:37:48.000000000 +0200
+++ lablgtk-2.16.0/configure	2013-08-21 11:16:50.707187151 +0200
@@ -4066,7 +4066,7 @@
       cat > conftest.ml << EOF
       open Raw
 EOF
-      if $CAMLC -c -I "${LABLGLDIR:=+lablGL}" conftest.ml > /dev/null 2>&1 ; then
+      if $CAMLC -c -I "$LABLGLDIR" conftest.ml > /dev/null 2>&1 ; then
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: $LABLGLDIR" >&5
 $as_echo "$LABLGLDIR" >&6; }
       else
