diff -u -r ocamlgraph-1.8.3/Makefile.in ocamlgraph-new/Makefile.in
--- Makefile.in	2013-04-17 20:30:17.000000000 +0200
+++ Makefile.in.new	2013-09-02 17:29:25.225954859 +0200
@@ -372,8 +372,10 @@
 	$(OCAMLFIND) install ocamlgraph META \
 		$(SRCDIR)/*.mli $(VIEWER_DIR)/*.mli $(DGRAPH_DIR)/*.mli \
 		graph$(LIBEXT) graph.cmx graph.cmo graph.cmi $(CMA) $(CMXA) \
-		$(VIEWER_CMXLIB) $(VIEWER_CMOLIB) $(DGRAPH_CMXLIB) \
-		$(DGRAPH_CMOLIB)
+		$(VIEWER_CMXLIB) $(VIEWER_CMOLIB) $(VIEWER_CMILIB) \
+                $(VIEWER_CMXLIB:.cmx=.o) \
+                $(DGRAPH_CMXLIB) $(DGRAPH_CMOLIB) $(DGRAPH_CMILIB) \
+                $(DGRAPH_CMXLIB:.cmx=.o)
 else
 	$(OCAMLFIND) install ocamlgraph META \
 		$(SRCDIR)/*.mli $(VIEWER_DIR)/*.mli $(DGRAPH_DIR)/*.mli \
