diff -C 2 -r ocp-indent-0.6.0.orig/src/ocp-indent.ocp ocp-indent-0.6.0/src/ocp-indent.ocp
*** ocp-indent-0.6.0.orig/src/ocp-indent.ocp	2013-01-21 12:27:47.000000000 +0100
--- ocp-indent-0.6.0/src/ocp-indent.ocp	2016-02-25 13:02:14.052188395 +0100
***************
*** 15,18 ****
     ]
     requires = [ ]
!    comp += [ "-g" "-w" "+A-4-9" "-warn-error" ]
  end
--- 15,18 ----
     ]
     requires = [ ]
!    comp += [ "-g" "-w" "+A-4-9" ]
  end
