diff --git a/src/OMakefile b/src/OMakefile
index ece98cd..242176a 100644
--- a/src/OMakefile
+++ b/src/OMakefile
@@ -10,7 +10,7 @@ section
     leveldb.cma leveldb.cmxa leveldb.a libcamlleveldb.a dllcamlleveldb.so: \
             leveldb_stubs.o levelDB.cmo levelDB.cmx
         ocamlmklib \
-            -oc camlleveldb -o leveldb -lsnappy -lleveldb \
+            -oc camlleveldb -o leveldb -lsnappy -lleveldb -lstdc++ -lsnappy -lm \
             leveldb_stubs.o levelDB.cmo levelDB.cmx
 
 .DEFAULT: leveldb.cma leveldb.cmxa
