From 44baaf74783974ac002b9f77a52a748450652afe Mon Sep 17 00:00:00 2001
From: Jean-Christophe Filliatre <Jean-Christophe.Filliatre@lri.fr>
Date: Fri, 23 Mar 2018 13:38:46 +0100
Subject: [PATCH] make uninstall

---
 CHANGES     | 2 ++
 Makefile.in | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/Makefile.in b/Makefile.in
index f62b61d..dfeb55a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -95,6 +95,11 @@ install: install-indep
 		cp bib2bib.byte $(BINDIR)/bib2bib ; \
 	fi
 
+uninstall:
+	rm -f $(BINDIR)/aux2bib $(BINDIR)/bibtex2html $(BINDIR)/bib2bib
+	rm -f $(MANDIR)/man1/aux2bib.1 $(MANDIR)/man1/bibtex2html.1 \
+	  $(MANDIR)/man1/bib2bib.1
+
 install-byte: install-indep
 	cp bibtex2html.byte $(BINDIR)/bibtex2html
 	cp bib2bib.byte $(BINDIR)/bib2bib
-- 
2.16.2

