diff -ru regstab.2.0.0/myocamlbuild.ml regstab.2.0.0/myocamlbuild.ml
--- regstab.2.0.0/myocamlbuild.ml	2012-05-04 05:14:37.000000000 +0200
+++ regstab.2.0.0/myocamlbuild.ml	2012-10-05 18:14:59.193868570 +0200
@@ -55,7 +55,7 @@
   dirs#man/"man1"/"sch2ltl.opt.1" , "man"/"man1"/"sch2ltl.opt.1";
   dirs#doc/"manual.pdf"           , "doc"/pkgbase;
   dirs#doc/"developer"            , "doc"/pkgbase/"developer_doc";
-  dirs#tools                      , "share"/pkgbase;
+  dirs#tools                      , "lib"/pkgbase/"share"/pkgbase;
   dirs#ex                         , "doc"/pkgbase;
   dirs#bin/"regstab"              , "bin"/"regstab";
   dirs#bin/"regstab.opt"          , "bin"/"regstab.opt";
