diff --git a/opam.install b/opam.install
index 92f49c7..89e06a7 100644
--- a/opam.install
+++ b/opam.install
@@ -25,5 +25,6 @@ lib: [
   "_obuild/opam-lib/opamPath.cmi"
   "_obuild/opam-lib/opamProcess.cmi"
   "_obuild/opam-lib/opamRepository.cmi"
+  "_obuild/opam-lib/opamRepositoryName.cmi"
   "_obuild/opam-lib/opamRsync.cmi"
   "_obuild/opam-lib/opamSolver.cmi"
   "_obuild/opam-lib/opamSystem.cmi"