diff --git a/Makefile.in b/Makefile.in
index ed3e41d..5a1549c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -46,7 +46,6 @@ OCAMLOPT = @OCAMLOPT@
 OCAMLDEP = @OCAMLDEP@
 OCAMLLEX = @OCAMLLEX@
 MENHIR = menhir # developers should have menhir installed
-OCAMLLIB = @OCAMLLIB@
 OCAMLBEST= @OCAMLBEST@
 OCAMLVERSION = @OCAMLVERSION@
 OCAMLWEB = @OCAMLWEB@
@@ -141,7 +140,7 @@ queens.opt: $(CMXA) queens.cmx
 # installation
 ##############
 
-INSTALL_LIBDIR=$(OCAMLLIB)/combine
+INSTALL_LIBDIR=$(prefix)/lib/combine
 
 install-indep:
 	mkdir -p $(BINDIR)
