diff --git a/Makefile.users b/Makefile.users
index 63e7bf2..294b510 100644
--- a/Makefile.users
+++ b/Makefile.users
@@ -177,15 +177,15 @@ lib/util/config.ml: config.status
 	@echo "let mandir = \""$(MANDIR)"\"" >> lib/util/config.ml
 
 lib/util/cmdline_parser.ml: config.status
-	@echo "(* parse_args () should be called as soon as possible (put earlier" > lib/util/cmdline_parser.ml
-	@echo "   when linking) because some choices during compilation depend on" >> lib/util/cmdline_parser.ml
-	@echo "   given options (e.g. dfs-sat)" >> lib/util/cmdline_parser.ml
-	@echo "\n   -> The following call:" >> lib/util/cmdline_parser.ml
-	@echo "   1 - should be kept when builtind Alt-Ergo tools" >> lib/util/cmdline_parser.ml
-	@echo "   2 - should be removed to build and use only the library." >> lib/util/cmdline_parser.ml
-	@echo "\n   Alternatively to 2, one can just remove cmdline_parser.cmo/cmx in" >> lib/util/cmdline_parser.ml
-	@echo "   Makefile.users *)" >> lib/util/cmdline_parser.ml
-	@echo "\nlet () = Options.parse_cmdline_arguments ()" >> lib/util/cmdline_parser.ml
+	@printf "(* parse_args () should be called as soon as possible (put earlier" > lib/util/cmdline_parser.ml
+	@printf "   when linking) because some choices during compilation depend on" >> lib/util/cmdline_parser.ml
+	@printf "   given options (e.g. dfs-sat)" >> lib/util/cmdline_parser.ml
+	@printf "\n   -> The following call:" >> lib/util/cmdline_parser.ml
+	@printf "   1 - should be kept when builtind Alt-Ergo tools" >> lib/util/cmdline_parser.ml
+	@printf "   2 - should be removed to build and use only the library." >> lib/util/cmdline_parser.ml
+	@printf "\n   Alternatively to 2, one can just remove cmdline_parser.cmo/cmx in" >> lib/util/cmdline_parser.ml
+	@printf "   Makefile.users *)" >> lib/util/cmdline_parser.ml
+	@printf "\nlet () = Options.parse_cmdline_arguments ()" >> lib/util/cmdline_parser.ml
 
 META: config.status
 	@echo "description = \"API of Alt-Ergo: An automatic theorem prover dedicated to program verification\"" > META
