--- a/configure	2016-07-22 21:53:10.872919000 +0200
+++ b/configure	2016-07-22 21:55:23.788871000 +0200
@@ -340,12 +340,8 @@
 if test "$gmp" = 'gmp' -o "$gmp" = 'auto'; then
     checkinc gmp.h
     if test $? -eq 1; then
-        checklib gmp
-        if test $? -eq 1; then 
-            gmp='OK'
-            cclib="$cclib -lgmp"
-            ccdef="-DHAS_GMP $ccdef"
-        fi
+        gmp='OK'
+        ccdef="-DHAS_GMP $ccdef"
     fi
 fi
 if test "$gmp" = 'mpir' -o "$gmp" = 'auto'; then
