diff --git a/configure b/configure
index 73cd90a..9e5fcc1 100755
--- a/configure
+++ b/configure
@@ -76,10 +76,10 @@ done
 echo -n "\
 BINDIR = $BINDIR
 MAN1DIR = $MANDIR/man1
-OCAMLC = $OCAMLC -w Ae -warn-error A
-OCAMLOPT = $OCAMLOPT -w Ae -warn-error A
+OCAMLC = $OCAMLC -w Ae
+OCAMLOPT = $OCAMLOPT -w Ae
 OCAMLC_UNSAFE = $OCAMLC
-OCAMLBUILD = $OCAMLBUILD -cflags -w,Ae,-warn-error,A -lflags -w,Ae,-warn-error,A -no-links -classic-display
+OCAMLBUILD = $OCAMLBUILD -cflags -w,Ae -lflags -w,Ae -no-links -classic-display
 BIN = $PWD/bin
 ETC = $PWD/etc
 DIST = $PWD/dist
