diff --git a/ocaml/zeroinstall/gui.ml b/ocaml/zeroinstall/gui.ml
index fa92430..8f15419 100644
--- a/ocaml/zeroinstall/gui.ml
+++ b/ocaml/zeroinstall/gui.ml
@@ -359,6 +359,8 @@ let try_get_gui config ~use_gui =
           let plugin_path =
             let sys_lib = Filename.dirname bindir +/ "lib" in
             U.first_match check_plugin_dir [
+              (* 0install lib dir *)
+              sys_lib +/ "0install";
               (* Is 0install is installed as distro package? *)
               sys_lib +/ "0install.net";
               (* Are we running via 0install? *)
