diff -ru extlib.1.5.2/Makefile extlib.1.5.2/Makefile
--- extlib.1.5.2/Makefile	2011-08-06 16:56:39.000000000 +0200
+++ extlib.1.5.2/Makefile	2012-10-05 18:15:17.734428987 +0200
@@ -20,7 +20,7 @@
 	cp odoc_style.css doc/style.css
 
 install:
-	ocamlfind install extlib META *.cmi *.cma $(MLI) $(wildcard *.cmxa) $(wildcard *.a)
+	ocamlfind install extlib META extLib.cmxs *.cmi *.cma $(MLI) $(wildcard *.cmxa) $(wildcard *.a)
 
 uninstall:
 	ocamlfind remove extlib
