diff --git a/Makefile b/Makefile
index 4a8ce17..3a184a1 100644
--- a/Makefile
+++ b/Makefile
@@ -89,10 +89,11 @@ install: FORCE
 	  $(PATH_OCAMLFIND) query $(PROJECT_NAME) && $(PATH_OCAMLFIND) remove $(PROJECT_NAME) || true; \
 	  $(PATH_OCAMLFIND) install $(PROJECT_NAME) META -optional \
 	    $(PATH_BUILD)/$(PROJECT_NAME)_pp.cmo \
-	    $(PATH_BUILD)/src/$(PROJECT_NAME)Thread.cm* \
-	    $(PATH_BUILD)/src/$(PROJECT_NAME)Thread.o \
-	    $(PATH_BUILD)/src/$(PROJECT_NAME)Thread.jo \
+	    $(PATH_BUILD)/src/threads/$(PROJECT_NAME)Thread.cm* \
+	    $(PATH_BUILD)/src/threads/$(PROJECT_NAME)Thread.o \
+	    $(PATH_BUILD)/src/threads/$(PROJECT_NAME)Thread.jo \
 	    $(PATH_BUILD)/$(PROJECT_NAME).a \
+	    $(PATH_BUILD)/$(PROJECT_NAME).o \
 	    $(PATH_BUILD)/$(PROJECT_NAME).cma \
 	    $(PATH_BUILD)/$(PROJECT_NAME).cmi \
 	    $(PATH_BUILD)/$(PROJECT_NAME).cmo \
@@ -106,7 +107,7 @@ install: FORCE
 	    cp $(PATH_BUILD)/$(PROJECT_NAME)_pp.cmo $(PATH_INSTALL); \
 	  fi; \
 	  for ext in cmi cmo cmx o cmj jo; do \
-	    test -f $(PATH_BUILD)/src/$(PROJECT_NAME)Thread.$$ext && cp $(PATH_BUILD)/src/$(PROJECT_NAME)Thread.$$ext $(PATH_INSTALL) || true; \
+	    test -f $(PATH_BUILD)/src/threads/$(PROJECT_NAME)Thread.$$ext && cp $(PATH_BUILD)/src/threads/$(PROJECT_NAME)Thread.$$ext $(PATH_INSTALL) || true; \
 	  done; \
 	  for ext in a cmi cmo cmx cma cmxa cmja ja; do \
 	    test -f $(PATH_BUILD)/$(PROJECT_NAME).$$ext && cp $(PATH_BUILD)/$(PROJECT_NAME).$$ext $(PATH_INSTALL) || true; \
