diff -u -r ott.0.24-orig/src/grammar_typecheck.ml ott.0.24/src/grammar_typecheck.ml
--- ott.0.24-orig/src/grammar_typecheck.ml	2014-01-30 18:54:47.000000000 +0100
+++ ott.0.24/src/grammar_typecheck.ml	2014-08-28 17:35:37.000000000 +0200
@@ -86,7 +86,7 @@
 avoid spurious conflicts.
 
 Generation of aux rules is controlled by a command-line option
--generate_aux_rules {true|false}, which one might (eg) set to false
+-generate_aux_rules {true | false}, which one might (eg) set to false
 for latex output and true for OCaml output.
 
 *)
