--- eliom-3.0.3/src/server/Makefile.orig	2013-09-01 23:03:58.000000000 +0100
+++ eliom-3.0.3/src/server/Makefile	2013-09-01 23:04:17.000000000 +0100
@@ -158,7 +158,7 @@
 ifneq (${DEPEND},no)
 
 ## Ugly side effect to force ${PREDEP} creation before dependencies calculation
-dummy_var := $(shell $(MAKE) DEPEND=no ${PREDEP} >/tmp/log 2>&1)
+dummy_var := $(shell $(MAKE) DEPEND=no ${PREDEP} >/dev/null 2>&1)
 
 DEPS := $(patsubst %,.%.deps,${COMMON_FILES}) \
 	$(patsubst %,.%.deps,${COMMON_PFILES}) \
