From 3c64229844630eefab8df15db1626d6fb4f2ee9a Mon Sep 17 00:00:00 2001
From: Kate <kit.ty.kate@disroot.org>
Date: Mon, 15 Mar 2021 21:02:45 +0000
Subject: [PATCH] Fix BSD support

---
 tools/extract_args/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/extract_args/Makefile b/tools/extract_args/Makefile
index d6c80a1..4ba6aeb 100644
--- a/tools/extract_args/Makefile
+++ b/tools/extract_args/Makefile
@@ -1,7 +1,7 @@
 all: extract_args
 
 extract_args.ml: extract_args.mll
-	ocamllex -o $@ $<
+	ocamllex -o extract_args.ml extract_args.mll
 
 extract_args: extract_args.ml
 	ocamlc -o extract_args extract_args.ml
-- 
2.30.2

