From de6f4ab737b0f83f945845967ac4be867650a44c Mon Sep 17 00:00:00 2001
From: Anil Madhavapeddy <anil@recoil.org>
Date: Sun, 21 Jun 2020 22:56:47 +0100
Subject: [PATCH] add `-fcommon` unconditionally to fix build on gcc10

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index 07b1c3503..3210157f3 100755
--- a/configure
+++ b/configure
@@ -350,6 +350,8 @@ case "$bytecc,$host" in
     bytecccompopts="-fno-defer-pop $gcc_warnings";;
 esac
 
+bytecccompopts="-fcommon $bytecccompopts"
+
 # Configure compiler to use in further tests
 
 cc="$bytecc -O $bytecclinkopts"
@@ -752,6 +754,7 @@ case "$arch,$nativecc,$system,$host_type" in
   *,gcc*,*,*)          nativecccompopts="$gcc_warnings";;
 esac
 
+nativecccompopts="-fcommon $nativecccompopts"
 asppprofflags='-DPROFILING'
 
 case "$arch,$model,$system" in
-- 
2.27.0

