--- myocamlbuild.ml.prev	2013-01-31 11:03:46.484655713 -0500
+++ myocamlbuild.ml	2013-01-31 11:03:49.639904031 -0500
@@ -587,7 +587,7 @@
 
     (* enable warnings; make sure the '@' character isn't in the beginning;
        ms-dos interprets that character specially *)
-    flag ["compile"; "ocaml"] (S [A "-w"; A "A@Aemr-28"; A "-strict-sequence" ])
+    flag ["compile"; "ocaml"] (S [A "-w"; A "Aemr-28"; A "-strict-sequence" ])
 ;;
 
 let dispatch = function
