diff -u -r atd-2.1.0/atdgen/test/bucklescript/dune fixed-atd-2.1.0/atdgen/test/bucklescript/dune
--- atd-2.1.0/atdgen/test/bucklescript/dune	2019-12-04 07:48:48.000000000 +0100
+++ fixed-atd-2.1.0/atdgen/test/bucklescript/dune	2020-02-24 16:01:40.682693115 +0100
@@ -32,16 +32,20 @@
 
 (alias
  (name runtest)
+ (package atdgen)
  (action (diff bucklespec_bs.expected.ml bucklespec_bs.ml)))
 
 (alias
  (name runtest)
+ (package atdgen)
  (action (diff bucklespec_bs.expected.mli bucklespec_bs.mli)))
 
 (alias
  (name runtest)
+ (package atdgen)
  (action (diff bucklespec_j.expected.ml bucklespec_j.ml)))
 
 (alias
  (name runtest)
+ (package atdgen)
  (action (run ./bucklespec_roundtrip.exe)))
diff -u -r atd-2.1.0/atdgen/test/spec_js/dune fixed-atd-2.1.0/atdgen/test/spec_js/dune
--- atd-2.1.0/atdgen/test/spec_js/dune	2019-12-04 07:48:48.000000000 +0100
+++ fixed-atd-2.1.0/atdgen/test/spec_js/dune	2020-02-24 15:16:05.827796462 +0100
@@ -34,6 +34,7 @@
 
 (alias
  (name runtest)
+ (package atdgen)
  (action (diff spec_j.expected.json spec_j.json)))
 
 (rule
@@ -42,4 +43,5 @@
 
 (alias
  (name runtest)
+ (package atdgen)
  (action (diff spec_bs.expected.json spec_bs.json)))
