commit 47581e8e3ebbd68fe05d130b764ddc86902ed214
Author: Anil Madhavapeddy <anil@recoil.org>
Date:   Sun Jun 21 22:56:47 2020 +0100

    add `-fcommon` unconditionally to fix build on gcc10

diff --git a/configure b/configure
index 3edb9fd24..af50adf00 100755
--- a/configure
+++ b/configure
@@ -429,6 +429,8 @@ case "$bytecc,$target" in
     bytecccompopts="-fno-defer-pop $gcc_warnings";;
 esac
 
+bytecccompopts="-fcommon $bytecccompopts"
+
 # Configure compiler to use in further tests
 
 cc="$bytecc -O $bytecclinkopts"
@@ -878,6 +880,7 @@ case "$arch,$nativecc,$system,$target" in
   *,gcc*,*,*)          nativecccompopts="$gcc_warnings";;
 esac
 
+nativecccompopts="-fcommon $nativecccompopts"
 asppprofflags='-DPROFILING'
 
 case "$arch,$system" in
