diff --git a/remake.cpp b/remake.cpp
index 3a6af80..d5ac424 100644
--- a/remake.cpp
+++ b/remake.cpp
@@ -450,7 +450,7 @@ enum { INVALID_SOCKET = -1 };
 extern char **environ;
 #endif
 
-#if defined(WINDOWS) || defined(MACOSX)
+# if (defined(WINDOWS) || defined(MACOSX)) && ! defined(MSG_NOSIGNAL)
 enum { MSG_NOSIGNAL = 0 };
 #endif
 
@@ -786,7 +786,7 @@ struct log
 	}
 };
 
-static log debug;
+static struct log debug;
 
 struct log_auto_close
 {
