From: Stephane Glondu <steph@glondu.net>
Date: Sun, 16 Jul 2023 13:26:38 +0200
Subject: Add gsl_include.sexp to deps of sf.ml*

---
 src/dune | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dune b/src/dune
index a6d8ce9..4fc42c0 100644
--- a/src/dune
+++ b/src/dune
@@ -77,6 +77,6 @@
 
 (rule
   (targets sf.mli sf.ml)
-  (deps config/do_sf.exe sf.mli.q)
+  (deps config/do_sf.exe gsl_include.sexp sf.mli.q)
   (action (run config/do_sf.exe))
 )
