diff --git a/enumerative.ml b/enumerative.ml
index 31e6054..1374cd7 100644
--- a/enumerative.ml
+++ b/enumerative.ml
@@ -285,7 +285,7 @@ let init_tables procs s =
   let ht = HT.create (nb_vars + nb_consts) in
   let i = ref 0 in
   Term.Set.iter (fun t -> HT.add ht t !i; incr i) var_terms;
-  let max_id_vars = !i - 1in
+  let max_id_vars = !i - 1 in
   let proc_ids = ref [] in
   let first_proc = !i in
   List.iter (fun t -> HT.add ht t !i; proc_ids := !i :: !proc_ids; incr i)
