--- lib/META.prev	2012-12-13 16:05:41.644185169 -0500
+++ lib/META	2012-12-13 16:06:15.268184976 -0500
@@ -3,7 +3,7 @@
 version = "108.08.00"
 description = "Jane Street Capital's standard library overlay"
 requires =
-"sexplib.syntax sexplib fieldslib.syntax fieldslib bin_prot bin_prot.syntax pa_ounit pa_pipebang core bigarray pcre res unix threads"
+" sexplib  fieldslib bin_prot oUnit core bigarray pcre res unix threads"
 archive(byte) = "core_extended.cma"
 archive(byte, plugin) = "core_extended.cma"
 archive(native) = "core_extended.cmxa"
