--- a/src/netstring/netencoding.mli	2014-03-17 23:07:25.757083686 +0100
+++ b/src/netstring/netencoding.mli	2014-03-17 23:05:15.801089492 +0100
@@ -120,7 +120,7 @@
 	 * to ensure that all output lines have a length <= 76 bytes.
 	 *
 	 * Note unsafe characters:
-	 *   As recommended by RFC 2045, the characters [!#$\@[]^`{|}~]
+	 *   As recommended by RFC 2045, the characters [!#$\@[]^`|{}~]
 	 *   and the double quotes
 	 *   are additionally represented as hex tokens.        
 	 *   Furthermore, the letter 'F' is considered as unsafe if it

