--- myocamlbuild.ml	2016-02-14 13:30:11.000000000 +0000
+++ myocamlbuild.ml	2016-02-14 13:33:23.000000000 +0000
@@ -484,15 +484,15 @@
           (["oasis_library_snappy_ccopt"; "compile"],
             [
                (OASISExpr.EBool true,
-                 S [A "-ccopt"; A "-x"; A "-ccopt"; A "c++"])
+                 S [A "-ccopt"; A "-x"; A "-ccopt"; A "c++"; A "-ccopt"; A "-I/usr/local/include"])
             ]);
           (["oasis_library_snappy_cclib"; "link"],
             [
                (OASISExpr.EBool true,
-                 S [A "-cclib"; A "-lstdc++"; A "-cclib"; A "-lsnappy"])
+                 S [A "-cclib"; A "-lstdc++"; A "-cclib"; A "-lsnappy"; A "-cclib"; A "-L/usr/local/lib"])
             ]);
           (["oasis_library_snappy_cclib"; "ocamlmklib"; "c"],
-            [(OASISExpr.EBool true, S [A "-lstdc++"; A "-lsnappy"])])
+            [(OASISExpr.EBool true, S [A "-lstdc++"; A "-L/usr/local/lib"; A "-lsnappy"])])
        ];
      includes = [("", ["src"])];
      }
