diff --git a/setup.ml b/setup.ml
index 1315322..48312bc 100644
--- a/setup.ml
+++ b/setup.ml
@@ -6154,7 +6154,9 @@ let setup () = BaseSetup.setup setup_t;;
 
 let search_compiler_libs () =
   prerr_endline "I: Searching for OCaml compiler libraries";
-  let stdlib = BaseEnv.var_get "standard_library" in
+  let stdlib =
+    OASISExec.run_read_one_line ~ctxt:!BaseContext.default
+      "ocamlfind" ["printconf"; "destdir"] in
   let ( / ) = Filename.concat in
   try
     List.find (fun path -> Sys.file_exists (path / "types.cmi") || Sys.file_exists (path / "typing" / "types.cmi")) [
