commit 11a5454ad16e0fa7c9ce74905609fbed9db18e1b
Author: Pierre Boutillier <pierre.boutillier@nomadic-labs.com>
Date:   Fri Oct 16 17:35:04 2020 +0200

    Build: tezos-stdlib-unix needs Re

diff --git a/src/lib_stdlib_unix/dune b/src/lib_stdlib_unix/dune
index 4539125cf5..bbfbd85a28 100644
--- a/src/lib_stdlib_unix/dune
+++ b/src/lib_stdlib_unix/dune
@@ -11,6 +11,7 @@
             tezos-stdlib
             lwt.unix
             ipaddr.unix
+            re
             ptime
             ptime.clock.os
             mtime
diff --git a/src/lib_stdlib_unix/tezos-stdlib-unix.opam b/src/lib_stdlib_unix/tezos-stdlib-unix.opam
index 9ecebc4440..30b5d93df1 100644
--- a/src/lib_stdlib_unix/tezos-stdlib-unix.opam
+++ b/src/lib_stdlib_unix/tezos-stdlib-unix.opam
@@ -13,6 +13,7 @@ depends: [
   "tezos-error-monad"
   "tezos-event-logging"
   "tezos-stdlib"
+  "re"
   "lwt" { >= "3.0.0" }
   "ptime" { >= "0.8.4" }
   "mtime" { >= "1.0.0" }
