diff --git a/configure b/configure
index 6b25ac3..3f7fc63 100755
--- a/configure
+++ b/configure
@@ -374,7 +374,9 @@ if [ -n "${NO_OCAMLFIND:-}" ]; then
     OCAMLFIND=
     echo-skip Disabled
 elif  [ -z "${OCAMLFIND:-}" ]; then
-    OCAMLFIND=$OCAMLOPT_DIR/ocamlfind
+    if ! OCAMLFIND=$(which ocamlfind); then
+        OCAMLFIND=$OCAMLOPT_DIR/ocamlfind
+    fi
     if [ ! -x "$OCAMLFIND" ]; then
         OCAMLFIND=$OCAMLLIB/../../bin/ocamlfind
     fi
