diff --git a/myocamlbuild.pre.ml b/myocamlbuild.pre.ml
index 24b1d3c..1bf6c8e 100644
--- a/myocamlbuild.pre.ml
+++ b/myocamlbuild.pre.ml
@@ -28,7 +28,7 @@ let _ =
     match event with
     | After_rules ->
         flag ["ocaml"; "compile"; "my_warnings"]
-          (S[A "-w"; A "@1..3@8..12@14..21@23..40-41@43"]);
+          (S[A "-w"; A "@1-2+3@8..12@14..21@23..40-41@43"]);
     | _ -> ()
   );
 ;;
