From ef04e8b8b440a7012f5eb21b55eb89d138758097 Mon Sep 17 00:00:00 2001
From: Etienne Millon <etienne@cryptosense.com>
Date: Thu, 1 Sep 2016 11:14:25 +0200
Subject: [PATCH 2/2] build_with_trunk

---
 zlibstubs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/zlibstubs.c b/zlibstubs.c
index 661729a..b4e6c61 100644
--- a/zlibstubs.c
+++ b/zlibstubs.c
@@ -17,6 +17,7 @@
 
 #include <stdint.h>
 #include <zlib.h>
+#include <stdint.h>
 
 #include <caml/mlvalues.h>
 #include <caml/alloc.h>
@@ -169,7 +170,7 @@ value camlzip_inflateEnd(value vzs)
 
 value camlzip_update_crc32(value crc, value buf, value pos, value len)
 {
-  return caml_copy_int32(crc32((uint32_t) Int32_val(crc), 
+  return caml_copy_int32(crc32((uint32_t) Int32_val(crc),
                           &Byte_u(buf, Long_val(pos)),
                           Long_val(len)));
 }
-- 
2.9.3

