From f87e2e4641faea4ba8d1d96e5ed914cb02b69254 Mon Sep 17 00:00:00 2001
From: Marcello Seri <marcello.seri@citrix.com>
Date: Wed, 21 Jun 2017 15:24:14 +0100
Subject: [PATCH] Fix _oasis to use xen-gnt-unix

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>
---
 _oasis | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/_oasis b/_oasis
index 12ba1263..d8683bce 100644
--- a/_oasis
+++ b/_oasis
@@ -59,7 +59,7 @@ Executable "xcp-rrdd"
     threads,
     ppx_deriving_rpc,
     io-page.unix,
-    xen-gnt.unix,
+    xen-gnt-unix,
     oclock,
     systemd,
     inotify
@@ -96,7 +96,7 @@ Executable "test_rrdd_monitor"
     threads,
     ppx_deriving_rpc,
     io-page.unix,
-    xen-gnt.unix,
+    xen-gnt-unix,
     oclock
 
 Test "test_rrdd_monitor"
-- 
2.11.0

