--- a/setup.ml	2019-04-16 14:57:33.467531887 +0100
+++ b/setup.ml	2019-04-16 15:06:59.227695456 +0100
@@ -7031,7 +7031,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3"
                              ]);
                            (OASISExpr.EAnd
@@ -7041,7 +7040,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-DACCELERATE";
                                 "-msse2";
@@ -7208,7 +7206,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-DNDEBUG";
                                 "${XEN_CFLAGS}"
@@ -7220,7 +7217,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-DNDEBUG";
                                 "${XEN_CFLAGS}";
--- a/myocamlbuild.ml	2019-04-16 14:57:38.307496074 +0100
+++ b/myocamlbuild.ml	2019-04-16 14:58:02.587317168 +0100
@@ -654,8 +654,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3"
                    ]);
                (OASISExpr.EAnd
@@ -670,8 +668,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-DACCELERATE";
@@ -702,8 +698,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-DNDEBUG";
@@ -722,8 +716,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-DNDEBUG";
