diff --git a/bindings/ocaml/Makefile b/bindings/ocaml/Makefile
index 44562fe8..13ca003 100644
--- a/bindings/ocaml/Makefile
+++ b/bindings/ocaml/Makefile
@@ -12,9 +12,9 @@ DIRS = llvm bitreader bitwriter irreader analysis target executionengine \
        transforms linker backends
 ExtraMakefiles = $(PROJ_OBJ_DIR)/Makefile.ocaml
 
-ocamldoc:
+build:
 	$(Verb) for i in $(DIRS) ; do \
-		$(MAKE) -C $$i ocamldoc; \
+		$(MAKE) -C $$i all-local; \
 	done
 
 include $(LEVEL)/Makefile.common
