diff --git a/scripts/mk_util.py b/scripts/mk_util.py
index 0177dc564..d5912cb24 100644
--- a/scripts/mk_util.py
+++ b/scripts/mk_util.py
@@ -2616,7 +2616,7 @@ def mk_config():
             CPPFLAGS = '%s -D_MP_INTERNAL' % CPPFLAGS
         if GIT_HASH:
             CPPFLAGS = '%s -DZ3GITHASH=%s' % (CPPFLAGS, GIT_HASH)
-        CXXFLAGS = '%s -std=c++20' % CXXFLAGS
+        CXXFLAGS = '%s -std=c++2a' % CXXFLAGS
         CXXFLAGS = '%s -fvisibility=hidden -fvisibility-inlines-hidden -c' % CXXFLAGS
         FPMATH = test_fpmath(CXX)
         CXXFLAGS = '%s %s' % (CXXFLAGS, FPMATH_FLAGS)
