diff -ru why3.0.73/configure why3.0.73/configure
--- why3.0.73/configure	2012-07-19 14:05:03.000000000 +0200
+++ why3.0.73/configure	2012-10-05 18:15:11.894252466 +0200
@@ -3464,7 +3464,7 @@
 # checking for lablgtksourceview2
 if test "$enable_ide" = yes ; then
    if test "$USEOCAMLFIND" = yes; then
-      LABLGTKSV2LIB=$(ocamlfind query lablgtksourceview2)
+      LABLGTKSV2LIB=$(ocamlfind query lablgtk2.sourceview2)
    fi
    if test -n "$LABLGTKSV2LIB";then
       echo "ocamlfind found lablgtksourceview2 in $LABLGTKSV2LIB"
