diff --git a/setup.ml b/setup.ml
index 795b218..36b36cd 100644
--- a/setup.ml
+++ b/setup.ml
@@ -5946,7 +5946,11 @@ let setup_t =
                      cs_plugin_data = [];
                      },
                    {
-                      bs_build = [(OASISExpr.EBool true, true)];
+                      bs_build =
+                        [
+                           (OASISExpr.EBool true, false);
+                           (OASISExpr.EFlag "tests", true)
+                        ];
                       bs_install = [(OASISExpr.EBool true, false)];
                       bs_path = "tests";
                       bs_compiled_object = Best;
@@ -5973,7 +5977,11 @@ let setup_t =
                      cs_plugin_data = [];
                      },
                    {
-                      bs_build = [(OASISExpr.EBool true, true)];
+                      bs_build =
+                        [
+                           (OASISExpr.EBool true, false);
+                           (OASISExpr.EFlag "tests", true)
+                        ];
                       bs_install = [(OASISExpr.EBool true, false)];
                       bs_path = "tests";
                       bs_compiled_object = Best;
@@ -6002,7 +6010,11 @@ let setup_t =
                      cs_plugin_data = [];
                      },
                    {
-                      bs_build = [(OASISExpr.EBool true, true)];
+                      bs_build =
+                        [
+                           (OASISExpr.EBool true, false);
+                           (OASISExpr.EFlag "tests", true)
+                        ];
                       bs_install = [(OASISExpr.EBool true, false)];
                       bs_path = "tests";
                       bs_compiled_object = Best;
