diff --git a/src_plugins/map/dune b/src_plugins/map/dune
--- a/src_plugins/map/dune
+++ b/src_plugins/map/dune
@@ -1,5 +1,5 @@
 (rule
- (deps ppx_deriving_map.cppo.ml)
+ (deps ../compat_macros.cppo ppx_deriving_map.cppo.ml)
  (targets ppx_deriving_map.ml)
  (action (run %{bin:cppo} -V OCAML:%{ocaml_version} %{deps} -o %{targets})))
 
diff --git a/src_plugins/map/ppx_deriving_map.cppo.ml b/src_plugins/map/ppx_deriving_map.cppo.ml
--- a/src_plugins/map/ppx_deriving_map.cppo.ml
+++ b/src_plugins/map/ppx_deriving_map.cppo.ml
@@ -1,5 +1,3 @@
-#include "../compat_macros.cppo"
-
 open Longident
 open Location
 open Asttypes
