diff --git a/str/META.template b/str/META.template
index b692995..961e89b 100644
--- a/str/META.template
+++ b/str/META.template
@@ -2,7 +2,7 @@ name = "mikmatch_str"
 version = "VERSION"
 description = "Pattern matching extended with regexps in Ocamllex syntax"
 
-requires = "camlp4 pcre unix"
+requires = "camlp4 str unix"
 requires(toploop) += "tophide"
 
 archive(syntax,toploop) = "pa_mikmatch_str.cma run_mikmatch_str.cma"
