diff --git a/src/import0.ml b/src/import0.ml
index 18d8a5c..bae0904 100644
--- a/src/import0.ml
+++ b/src/import0.ml
@@ -193,7 +193,7 @@ include Int_replace_polymorphic_compare
 let ( <.  ) (x : float) y = Poly.( <  ) x y
 let ( <=. ) (x : float) y = Poly.( <= ) x y
 let ( =.  ) (x : float) y = Poly.( =  ) x y
-let ( >.  ) (x : float) y = Poly.( >  ) x y
+let ( >:  ) (x : float) y = Poly.( >  ) x y
 let ( >=. ) (x : float) y = Poly.( >= ) x y
 
 (* This needs to be defined as an external so that the compiler can specialize it as a
