diff --git a/Makefile b/Makefile
index 4fcf156..c2571d0 100644
--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ install:
 
 find-install:
        ocamlfind install agrep \
-               agrep.cma agrep.cmi agrep.cmxa libagrep.a dllagrep.$(LIB_EXT) META
+               agrep.cma agrep.cmi agrep.cmxa agrep.a agrep.cmx libagrep.a dllagrep.$(LIB_EXT) META
 
 testagrep: testagrep.ml agrep.cma libagrep.a
        $(OCAMLC) -I . -custom -o $@ agrep.cma testagrep.ml
