diff --git a/lib/core_unix.ml b/lib/core_unix.ml
index 7eb2427..4dd1496 100644
--- a/lib/core_unix.ml
+++ b/lib/core_unix.ml
@@ -890,6 +890,7 @@ Unix.open_flag =
 | O_SYNC
 | O_RSYNC
 | O_SHARE_DELETE
+| O_CLOEXEC
 with sexp
 
 type file_perm = int with of_sexp
diff --git a/lib/core_unix.mli b/lib/core_unix.mli
index 890d7ec..a367a9b 100644
--- a/lib/core_unix.mli
+++ b/lib/core_unix.mli
@@ -305,6 +305,7 @@ type open_flag =
     | O_SYNC          (** Writes complete as `Synchronised I/O file integrity completion' *)
     | O_RSYNC         (** Reads complete as writes (depending on O_SYNC/O_DSYNC) *)
     | O_SHARE_DELETE  (** Windows only: allow the file to be deleted while still open *)
+    | O_CLOEXEC       (** Set the close-on-exec flag on the descriptor returned by {!openfile} *)
 with sexp
 
 (** The type of file access rights. *)

