diff --git a/framaCIRGen_src/build.sh b/framaCIRGen_src/build.sh
index cdd9fcd0..103d6f9c 100755
--- a/framaCIRGen_src/build.sh
+++ b/framaCIRGen_src/build.sh
@@ -1,4 +1,4 @@
-#! /usr/bin/bash
+#! /bin/sh
 ##########################################################################
 #                                                                        #
 #  This file is part of Frama-Clang                                      #
