From cc9431e3ee31774f0d37ab4b4f9e964e96482e98 Mon Sep 17 00:00:00 2001
From: OCamlPro-Iguernlala <mohamed.iguernlala@ocamlpro.com>
Date: Thu, 26 Apr 2018 19:09:41 +0200
Subject: [PATCH] fix uninstall-lib in Makefile

---
 Makefile.users | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.users b/Makefile.users
index fdad643..1055667 100644
--- a/Makefile.users
+++ b/Makefile.users
@@ -263,18 +263,18 @@ install-lib: lib META
 #	cp -f META $(LIBDIR)
 
 uninstall-lib:
-	ocamlfind remove alt-ergo-lib
+	ocamlfind remove alt-ergo
 
 install: install-ae install-gui install-fm-simplex install-lib
 
-uninstall: uninstall-lib
+uninstall:
 	@rm $(BINDIR)/$(NAME)$(EXE)
 	@rm $(MANDIR)/man1/alt-ergo.1
 	-@rm $(BINDIR)/$(GUINAME)$(EXE)
 	-@rm $(DATADIR)/gtksourceview-2.0/language-specs/alt-ergo.lang
 	@rm $(PLUGINSDIR)/fm-simplex-plugin.cm*
 	@rm $(PRELUDESDIR)/fpa-theory-2017-01-04-16h00.why
-#	-@rm $(LIBDIR)/$(LIBNAME).*
+	make uninstall-lib
 #	-@rm $(LIBDIR)/META
 #	-@rm $(LIBDIR)
 
-- 
2.14.1

