diff -C 2 -r ocp-indent.0.9.0.orig/src/ocp-indent.ocp ocp-indent.0.9.0/src/ocp-indent.ocp
*** ocp-indent.0.9.0.orig/src/ocp-indent.ocp	2016-02-25 13:28:02.504254590 +0100
--- ocp-indent.0.9.0/src/ocp-indent.ocp	2016-02-25 13:28:19.912255334 +0100
***************
*** 1,3 ****
! comp += [ "-g" "-w" "+A-4-9" "-warn-error" ]
  
  begin library "ocp-indent-lexer"
--- 1,3 ----
! comp += [ "-g" "-w" "+A-4-9" ]
  
  begin library "ocp-indent-lexer"
