diff -Naur camlidl-1.05/config/Makefile.unix camlidl-1.05-patched/config/Makefile.unix
--- camlidl-1.05/config/Makefile.unix	2002-04-22 12:50:46.000000000 +0100
+++ camlidl-1.05-patched/config/Makefile.unix	2013-04-09 13:43:56.751790572 +0100
@@ -19,9 +19,9 @@
 
 # How to invoke the C preprocessor
 # Works on most Unix systems:
-CPP=/lib/cpp
+CPP=cpp
 # Alternatives:
-# CPP=cpp
+# CPP=/lib/cpp
 # CPP=/usr/ccs/lib/cpp
 # CPP=gcc -x c -E
 
