diff --git a/lib/mustache.ml b/lib/mustache.ml
index 6a5a146..40fb611 100644
--- a/lib/mustache.ml
+++ b/lib/mustache.ml
@@ -5,6 +5,9 @@ module Str = Re_str
 module List = ListLabels
 module String = StringLabels
 
+let (|>) x f = f x
+let (@@) f x = f x
+
 exception Invalid_param of string with sexp
 exception Invalid_template of string with sexp
 
diff --git a/lib_test/test_mustache.ml b/lib_test/test_mustache.ml
index 516c1c0..a8242f3 100644
--- a/lib_test/test_mustache.ml
+++ b/lib_test/test_mustache.ml
@@ -2,6 +2,9 @@ open OUnit2
 open Printf
 open Sexplib.Std
 
+let (|>) x f = f x
+let (@@) f x = f x
+
 module List = ListLabels
 module String = StringLabels
 
