diff -ru unison.2.40.63/Makefile.OCaml unison.2.40.63/Makefile.OCaml
--- unison.2.40.63/Makefile.OCaml	2011-04-16 22:35:38.000000000 +0200
+++ unison.2.40.63/Makefile.OCaml	2012-10-05 18:15:10.918222957 +0200
@@ -285,14 +285,14 @@
 
 # Gtk GUI
 ifeq ($(UISTYLE), gtk)
-  CAMLFLAGS+=-I +lablgtk
+  CAMLFLAGS+=-I $(LABLGTKLIB)
   OCAMLOBJS+=pixmaps.cmo uigtk.cmo linkgtk.cmo
   OCAMLLIBS+=lablgtk.cma
 endif
 
 # Gtk2 GUI
 ifeq ($(UISTYLE), gtk2)
-  CAMLFLAGS+=-I +lablgtk2
+  CAMLFLAGS+=-I $(LABLGTK2LIB)
   OCAMLOBJS+=pixmaps.cmo uigtk2.cmo linkgtk2.cmo
   OCAMLLIBS+=lablgtk.cma
 endif
