diff -u -r ocp-ocamlres.0.3-orig/Makefile ocp-ocamlres.0.3/Makefile
--- ocp-ocamlres.0.3-orig/Makefile	2014-05-15 14:19:52.000000000 +0200
+++ ocp-ocamlres.0.3/Makefile	2014-08-28 15:48:37.000000000 +0200
@@ -76,7 +76,7 @@
 -include .depend
 .depend:
 	ocamlfind ocamldep -I src -package $(PACKAGES) $(LIB_ML) > $@
-	sed -i s/src/build/g $@
+	sed -i.old s/src/build/g $@
 
 $(BIN_CMO): build/ocplib-ocamlres.cma
 $(BIN_CMX): build/ocplib-ocamlres.cmxa
