diff -ru /tmp/opam-12040-3941/sqlite3-ocaml.1.6.3/test/Makefile /tmp/opam-12040-480/sqlite3-ocaml.1.6.3/test/Makefile
--- /tmp/opam-12040-3941/sqlite3-ocaml.1.6.3/test/Makefile	2012-05-20 04:01:44.000000000 +0200
+++ /tmp/opam-12040-480/sqlite3-ocaml.1.6.3/test/Makefile	2012-10-05 18:15:15.110349675 +0200
@@ -8,10 +8,10 @@
 opt: $(TESTS:%.ml=%.exe)
 
 %.bc: %.ml
-	ocamlc -w Ae -warn-error Ae -I .. -dllpath .. str.cma sqlite3.cma $< -o $@
+	ocamlc  -I .. -dllpath .. str.cma sqlite3.cma $< -o $@
 
 %.exe: %.ml
-	ocamlopt -w Ae -warn-error Ae -I .. str.cmxa sqlite3.cmxa $< -o $@
+	ocamlopt  -I .. str.cmxa sqlite3.cmxa $< -o $@
 
 clean:
 	rm -f \
