diff --git a/src/dune b/src/dune
index be85e03..e1c7ca9 100644
--- a/src/dune
+++ b/src/dune
@@ -23,7 +23,7 @@
  (action
   (bash "
    (../../../src/3rd-party/tokyocabinet/configure --disable-bzip --disable-zlib
-   make libtokyocabinet.so libtokyocabinet.a -j $(nproc)) > /dev/null 2>&1
+   make libtokyocabinet.so libtokyocabinet.a) > /dev/null 2>&1
    cp libtokyocabinet.so dlltc_stubs.so
    cp libtokyocabinet.a libtc_stubs.a ")
  )
