diff --git a/vendors/ocaml-lmdb/config/discover.ml b/vendors/ocaml-lmdb/config/discover.ml
index 1b0496ec98..52542cd24e 100644
--- a/vendors/ocaml-lmdb/config/discover.ml
+++ b/vendors/ocaml-lmdb/config/discover.ml
@@ -3,6 +3,7 @@ let () =
   let w = "-W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized" in
   let thread = "-pthread" in
   let opt = "-O2 -g" in
-  Printf.fprintf oc "(%s %s %s %s)" w thread opt
+  let freebsd_stuff = "-I/usr/local/include -D_WANT_SEMUN" in
+  Printf.fprintf oc "(%s %s %s %s %s)" w thread opt freebsd_stuff
     (if Sys.word_size = 32 then "-DMDB_VL32" else "") ;
   close_out oc
