From 9a7ec63d4993856ec4cf862f02fb5b051dd073c4 Mon Sep 17 00:00:00 2001
From: Rudi Grinberg <rudi.grinberg@gmail.com>
Date: Tue, 27 Feb 2018 02:52:58 +0700
Subject: [PATCH] fmt.tty is used by datakit-log

---
 src/datakit-log/jbuild | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/datakit-log/jbuild b/src/datakit-log/jbuild
index 86ed620..3a8465b 100644
--- a/src/datakit-log/jbuild
+++ b/src/datakit-log/jbuild
@@ -4,4 +4,4 @@
  ((name        datakit_log)
   (wrapped     false)
   (libraries   (cmdliner fmt logs.cli prometheus-app.unix mtime mtime.clock.os
-                win-eventlog asl logs.fmt))))
+                win-eventlog asl logs.fmt fmt.tty))))
-- 
2.16.3

