diff --git a/vendors/ocaml-lmdb/config/discover.ml b/vendors/ocaml-lmdb/config/discover.ml
index 1b0496ec98..d04ee8a76e 100644
--- a/vendors/ocaml-lmdb/config/discover.ml
+++ b/vendors/ocaml-lmdb/config/discover.ml
@@ -3,6 +3,8 @@ let () =
   let w = "-W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized" in
   let thread = "-pthread" in
   let opt = "-O2 -g" in
-  Printf.fprintf oc "(%s %s %s %s)" w thread opt
+  let includes = "-I/usr/local/include"
+  (* consistant with conf-gmp.2 opam package, useful at least for *bsd *) in
+  Printf.fprintf oc "(%s %s %s %s %s)" w thread opt includes
     (if Sys.word_size = 32 then "-DMDB_VL32" else "") ;
   close_out oc
