From 653eac9553482c6a57c1e55068fe86a0f5c1201d Mon Sep 17 00:00:00 2001
From: Marcello Seri <marcello.seri@citrix.com>
Date: Tue, 20 Jun 2017 11:02:23 +0100
Subject: [PATCH] Update mktemp template in configure to support busybox

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>
---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 5ad76e3..d9d2e21 100755
--- a/configure
+++ b/configure
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-D=$(mktemp -d /tmp/configure.XXXXX)
+D=$(mktemp -d /tmp/configure.XXXXXX)
 function cleanup {
   cd /
   rm -rf $D
-- 
2.11.0

