commit c6604df9aa68ede640fc07459099ef953f4520e7
author: Steven de Oliveira <de.oliveira.steven@gmail.com>
date:   Fri, 1 Jul 2022 10:38:36 +0200

	Dune fix

diff --git a/configure.ml b/configure.ml
index ad9489ec..1ccd4d61 100644
--- a/configure.ml
+++ b/configure.ml
@@ -139,26 +139,6 @@ let () =
     Format.eprintf "ERROR: Couldn't find dune in env@.";
     exit 1
 
-(* run dune to check that dependencies are installed *)
-let () =
-  let p_opt = match !pkg with
-    | "" -> ""
-    | s -> Format.asprintf "-p %s" s
-  in
-  let cmd = Format.asprintf
-      "dune external-lib-deps --display=quiet --missing %s @install"
-      p_opt
-  in
-  let ch = Unix.open_process_in cmd in
-  let _ = read_all ch in
-  let res = Unix.close_process_in ch in
-  match res with
-  | Unix.WEXITED 0 ->
-    Format.printf "All deps are installed.@."
-  | _ ->
-    (* dune already prints the missing libs on stderr *)
-    exit 2
-
 let () =
   Format.printf "Good to go !@."
 
