diff --git a/src/nethttpd/nethttpd_types.ml b/src/nethttpd/nethttpd_types.ml
index af538c9..84f1521 100644
--- a/src/nethttpd/nethttpd_types.ml
+++ b/src/nethttpd/nethttpd_types.ml
@@ -216,8 +216,6 @@ object(self)
 
   (* ---- The following is copied from Netcgi_env: ---- *)
 
-  method input_header = in_header
-
   method input_header_field ?default name =
     try in_header # field name
     with Not_found as nf ->
@@ -250,9 +248,6 @@ object(self)
   method input_content_type() =
     Mimestring.scan_mime_type_ep (self # input_header_field "CONTENT-TYPE") []
 
-  method output_header =
-    out_header
-
   method output_header_field ?default name =
     try out_header # field name
     with Not_found as nf ->
