diff --git a/src/lib_webassembly/util/chunked_byte_vector.ml b/src/lib_webassembly/util/chunked_byte_vector.ml
index 0a0e829786..1b494f05b6 100644
--- a/src/lib_webassembly/util/chunked_byte_vector.ml
+++ b/src/lib_webassembly/util/chunked_byte_vector.ml
@@ -26,7 +26,7 @@ module Chunk = struct
 
   let of_bytes bytes =
     let chunk = alloc () in
-    for i = 0 to Int.max (Int64.to_int size) (Bytes.length bytes) - 1 do
+    for i = 0 to max (Int64.to_int size) (Bytes.length bytes) - 1 do
       Array1.set chunk i (Char.code (Bytes.get bytes i))
     done ;
     chunk
