diff --git a/configure.ac b/configure.ac
index d73d724..3653943 100644
--- a/configure.ac
+++ b/configure.ac
@@ -208,7 +208,7 @@ if test "$HAS_BENCHMARK" = "yes" ; then
 fi
 
 AS_IF([test "$OCAMLBEST" = "opt"],[OCAMLBESTCC=$OCAMLOPT],[OCAMLBESTCC=$OCAMLC])
-OCAML_CC="$($OCAMLBESTCC -config | fgrep native_c_compiler | sed -e "s/native_c_compiler: \(.*\) .*/\1/")"
+OCAML_CC="$($OCAMLBESTCC -config | grep -P "^native_c_compiler:" | sed -e "s/native_c_compiler: \(.*\) .*/\1/")"
 AC_PROG_CC(["${OCAML_CC}" gcc cl cc])
 
 AC_HEADER_STDC
@@ -254,7 +254,7 @@ if test "$HAS_RPM5" = "yes"; then
   HAS_RPM=yes
 fi
 
-OCAML_SYSTEM="$($OCAMLBESTCC -config | fgrep system | sed -e "s/system: \(.*\)/\1/")"
+OCAML_SYSTEM="$($OCAMLBESTCC -config | grep -P "^system:" | sed -e "s/system: \(.*\)/\1/")"
 AC_SUBST(OCAML_SYSTEM)
 
 AS_IF([test "${OCAML_OS_TYPE}" = "Win32"],[
