diff --git a/_tags b/_tags
index 2182dc8..4f6ad08 100644
--- a/_tags
+++ b/_tags
@@ -5,7 +5,7 @@ true: bin_annot, safe_string, thread
 <support/consts.{ml,native,byte}> : custom, link_consts_stub
 
 <src> : include
-<src/tsdl.{ml,mli}> : package(ctypes), thread
+<src/tsdl.{ml,mli}> : package(ctypes), package(ctypes.foreign), thread
 <src/tsdl.ml> : sdl_consts
 <src/tsdl.{cma,cmxa,cmxs}> : custom, thread, \
  package(ctypes), package(ctypes.foreign), use_sdl2
