diff --git a/src/lbfgs.mli b/src/lbfgs.mli
index 7b432a7..0049f01 100644
--- a/src/lbfgs.mli
+++ b/src/lbfgs.mli
@@ -106,13 +106,13 @@ sig
 
     @param factr tolerance in the termination test for the algorithm.
     The iteration will stop when
-    [(f^k - f^{k+1})/max{|f^k|,|f^{k+1}|,1} <= factr*epsilon_float].
+    [(f^k - f^{k+1})/max{ |f^k|,|f^{k+1}|, 1 } <= factr*epsilon_float].
     Set e.g. [factr] to [1e12] for low accuracy, [1e7] for moderate
     accuracy and [1e1] for extremely high accuracy.  Setting [factr] to
     [0.] suppresses this termination test.  Default: [1e7].
 
     @param pgtol The iteration will stop when
-    [max{|proj g_i| : i = 1,..., n} <= pgtol]
+    [max{ |proj g_i| : i = 1,..., n} <= pgtol]
     where [proj g_i] is the ith component of the projected gradient.
     Setting [pgtol] to [0.] suppresses this termination test.
     Default: [1e-5].
