diff -crB libsvm-ocaml-0.8.3.orig/lib/libsvm.ml libsvm-ocaml-0.8.3/lib/libsvm.ml
*** libsvm-ocaml-0.8.3.orig/lib/libsvm.ml	2013-03-01 23:04:00.689977974 +0100
--- libsvm-ocaml-0.8.3/lib/libsvm.ml	2013-03-01 23:04:56.694277823 +0100
***************
*** 543,553 ****
      let sum_yy = ref 0. in
      let sum_xy = ref 0. in
      Array.iter2_exn array_x array_y ~f:(fun x y ->
!       sum_x  := !sum_x +. x;
!       sum_y  := !sum_y +. y;
!       sum_xx := x *. x;
!       sum_yy := y *. y;
!       sum_xy := x *. y;
      );
      let sqr x = x *. x in
      let l = float (Vec.dim expected) in
--- 543,553 ----
      let sum_yy = ref 0. in
      let sum_xy = ref 0. in
      Array.iter2_exn array_x array_y ~f:(fun x y ->
!       sum_x  := !sum_x  +. x;
!       sum_y  := !sum_y  +. y;
!       sum_xx := !sum_xx +. x *. x;
!       sum_yy := !sum_yy +. y *. y;
!       sum_xy := !sum_xy +. x *. y;
      );
      let sqr x = x *. x in
      let l = float (Vec.dim expected) in
