diff -crB libsvm-ocaml-0.8.3.orig/examples/svm_cli.ml libsvm-ocaml-0.8.3/examples/svm_cli.ml
*** libsvm-ocaml-0.8.3.orig/examples/svm_cli.ml	2013-03-01 21:25:49.894623050 +0100
--- libsvm-ocaml-0.8.3/examples/svm_cli.ml	2013-03-01 21:26:19.808249078 +0100
***************
*** 13,25 ****
  
  let read_scale_parameters file =
    let conv_line line =
!     sscanf line "%d %g %g" (fun _index min max -> min, max)
    in
    In_channel.with_file file ~f:(fun ic ->
      match In_channel.input_line ic with
      | None -> failwith "no data"
      | Some line ->
!       let lower, upper = sscanf line "%g %g" Tuple2.create in
        let lines = In_channel.input_lines ic in
        let min_feats, max_feats =
          List.map lines ~f:conv_line
--- 13,25 ----
  
  let read_scale_parameters file =
    let conv_line line =
!     Scanf.sscanf line "%d %g %g" (fun _index min max -> min, max)
    in
    In_channel.with_file file ~f:(fun ic ->
      match In_channel.input_line ic with
      | None -> failwith "no data"
      | Some line ->
!       let lower, upper = Scanf.sscanf line "%g %g" Tuple2.create in
        let lines = In_channel.input_lines ic in
        let min_feats, max_feats =
          List.map lines ~f:conv_line
