diff --git a/code/src/nettls-gnutls/configure b/code/src/nettls-gnutls/configure
index 247ee47..65833db 100755
--- a/src/nettls-gnutls/configure
+++ b/src/nettls-gnutls/configure
@@ -9,7 +9,10 @@
 have_gnutls=0
 
 if [ -z "$PKG_CONFIG" ]; then
-    PKG_CONFIG="pkg-config"
+    PKG_CONFIG="pkg-config"
+    # Workaround issue on OpenBSD
+    export PKG_CONFIG_ALLOW_SYSTEM_LIBS=1
+    export PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1
 fi
 
 if [ -z "$GNUTLS_LIBS" ]; then

