diff --git a/code/src/nettls-gnutls/configure b/code/src/nettls-gnutls/configure
index 247ee47..84fa991 100755
--- a/code/src/nettls-gnutls/configure
+++ b/code/src/nettls-gnutls/configure
@@ -9,7 +9,12 @@
 have_gnutls=0
 
 if [ -z "$PKG_CONFIG" ]; then
+  unamestr=$(uname)
+  if [ "$unamestr" = "OpenBSD" ]; then
+    PKG_CONFIG="pkg-config --keep-system-cflags --keep-system-libs"
+  else
     PKG_CONFIG="pkg-config"
+  fi
 fi
 
 if [ -z "$GNUTLS_LIBS" ]; then

