commit 2d04838fcff65c494316562bd415ec98e4a82703
Author: Pierre Boutillier <pierre.boutillier@nomadic-labs.com>
Date:   Fri Oct 16 18:05:19 2020 +0200

    Opam: clic requires re

diff --git a/src/lib_clic/dune b/src/lib_clic/dune
index 421a42c891..22f96d5f10 100644
--- a/src/lib_clic/dune
+++ b/src/lib_clic/dune
@@ -5,6 +5,7 @@
                    -open Tezos_error_monad))
  (libraries tezos-stdlib
             lwt
+            re
             tezos-error-monad))
 
 (rule
