diff --git a/Makefile.common b/Makefile.common
index d3dadef1c5..e23f3a4dff 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -35,7 +35,7 @@ _DBUILD_DEPS+=$(FLOCK)
 
 $(FLOCK): tools/flock/coq_flock.ml tools/flock/flock.c
 	$(SHOW)'DUNE      $@'
-	$(HIDE)dune build $@
+	$(HIDE)dune build --root . $@
 else
 FLOCK:=flock
 endif
diff --git a/Makefile.install b/Makefile.install
index 18a3e39177..c8df10a3c6 100644
--- a/Makefile.install
+++ b/Makefile.install
@@ -53,8 +53,8 @@ endif
 # We ought to improve this, as of today the case of an opam-built dune
 # installing to a global prefix such as /usr/local/ may not follow the FHS
 install-dune: $(BCONTEXT)/coq-core.install $(BCONTEXT)/coqide-server.install
-	dune install --display quiet $(DESTDIRARG) --mandir="$(MANDIR)" --prefix="$(COQPREFIX)" coq-core
-	dune install --display quiet $(DESTDIRARG) --mandir="$(MANDIR)" --prefix="$(COQPREFIX)" coqide-server
+	dune install $(_DOPT) $(DESTDIRARG) --mandir="$(MANDIR)" --prefix="$(COQPREFIX)" coq-core
+	dune install $(_DOPT) $(DESTDIRARG) --mandir="$(MANDIR)" --prefix="$(COQPREFIX)" coqide-server
 
 install-coq: install-dune install-library
 
@@ -87,7 +87,7 @@ ifeq ($(HASCOQIDE),no)
 install-coqide:
 else
 install-coqide: $(BCONTEXT)/coqide.install
-	dune install --display quiet $(DESTDIRARG) --mandir="$(MANDIR)" --prefix="$(COQPREFIX)" coqide
+	dune install $(_DOPT) $(DESTDIRARG) --mandir="$(MANDIR)" --prefix="$(COQPREFIX)" coqide
 endif
 
 # For emacs:
