commit 52dac4419fe23c7bbfc27bee060ed38d0153be98
Author: ygrek <ygrek@autistici.org>
Date:   Sun Nov 8 18:07:08 2015 -0800

    do not build examples by default (fix #10)

diff --git a/Makefile.in b/Makefile.in
index 3f93a2f..9374397 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -49,7 +49,7 @@ endif
 
 all:
 		@$(MAKE) depend
-		@$(MAKE) targets
+		@$(MAKE) $(TARGETS)
 
 build: all
 
@@ -75,8 +75,6 @@ else
 		ocamldoc -html -d doc curl.mli
 endif
 
-targets:	$(TARGETS) examples
-
 examples:
 		(cd examples; $(MAKE))
 
