diff -ur acgtk.1.1/src/datalog.solver/Makefile.in acgtk.1.1_patched/src/datalog.solver/Makefile.in
--- acgtk.1.1/src/datalog.solver/Makefile.in	2014-09-05 21:00:49.000000000 +0200
+++ acgtk.1.1_patched/src/datalog.solver/Makefile.in	2015-01-14 15:57:45.190596042 +0100
@@ -33,7 +33,7 @@
 # ex. -I  +campl4
 LIBDIR +=
 
-WARNINGS = -w yzd -warn-error A
+WARNINGS = -w yzd
 
 # Directories to which the current source files depend on
 PREVIOUS_DIRS = 
diff -ur acgtk.1.1/src/Makefile.master.in acgtk.1.1_patched/src/Makefile.master.in
--- acgtk.1.1/src/Makefile.master.in	2014-02-09 08:41:32.000000000 +0100
+++ acgtk.1.1_patched/src/Makefile.master.in	2015-01-14 15:58:01.026470929 +0100
@@ -21,7 +21,7 @@
 OCAMLCPOPT = -p fm
 
 # All warnings are treated as errors
-WARNINGS = @OCAML09WARNINGS@ -warn-error A
+WARNINGS = @OCAML09WARNINGS@
 COMMONFLAGS= $(WARNINGS) @TYPES@
 BYTEFLAGS = $(COMMONFLAGS) $(DEBUGFLAG)
 OPTFLAGS = $(COMMONFLAGS)
