diff --git a/configure b/configure
index 177e9547..f8a56fe0 100755
--- a/configure
+++ b/configure
@@ -4925,7 +4925,7 @@ printf %s "checking lablGL directory... " >&6; }
       cat > conftest.ml << EOF
       open Raw
 EOF
-      if $CAMLC -c -I "${LABLGLDIR:=+lablGL}" conftest.ml > /dev/null 2>&1 ; then
+      if $CAMLC -c -I "$LABLGLDIR" conftest.ml > /dev/null 2>&1 ; then
         { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $LABLGLDIR" >&5
 printf "%s\n" "$LABLGLDIR" >&6; }
       else
