commit f5f6c9bb104525a44db53d1230fdf7ffab1345de
Author: François Bobot <francois.bobot@cea.fr>
Date:   Sun Jun 19 20:10:41 2022 +0200

    Remove large integer constant in colibrics

diff --git a/src/bin/options.ml b/src/bin/options.ml
index 721859b3..bd5a281c 100644
--- a/src/bin/options.ml
+++ b/src/bin/options.ml
@@ -152,12 +152,10 @@ let parse_time arg =
 let size_string f =
   let n = int_of_float f in
   let aux n div = n / div, n mod div in
-  let n_tera, n = aux n 1_000_000_000_000 in
   let n_giga, n = aux n 1_000_000_000 in
   let n_mega, n = aux n 1_000_000 in
   let n_kilo, n = aux n 1_000 in
   let print_aux s n = if n <> 0 then (string_of_int n) ^ s else "" in
-  (print_aux "To" n_tera) ^
   (print_aux "Go" n_giga) ^
   (print_aux "Mo" n_mega) ^
   (print_aux "ko" n_kilo) ^
