--- dtd.ml.prev	2013-01-04 20:57:33.174276611 +0100
+++ dtd.ml	2013-01-04 20:58:00.722277268 +0100
@@ -267,7 +267,7 @@
 exception TmpResult of dtd_result
 
 let prove_child dtd tag = 
-	trace dtd tag;
+	(* trace dtd tag; *)
 	match dtd.current with
 	| DTDEmpty -> raise (Prove_error EmptyExpected)
 	| DTDAny -> ()
@@ -505,4 +505,4 @@
 		sprintf "<!ELEMENT %s %s>" tag (etype_to_string etype)
 
 ;;
-to_string_ref := to_string
\ No newline at end of file
+to_string_ref := to_string
