diff --git a/tests/merlin-wrapper b/tests/merlin-wrapper
index 43f050e1..5b8d8184 100755
--- a/tests/merlin-wrapper
+++ b/tests/merlin-wrapper
@@ -5,6 +5,5 @@ export PATH=$(dirname dot-merlin-reader):$PATH
 touch .merlin && ocamlmerlin "$@" \
     | jq 'del(.timing)' \
     | sed -e 's:"[^"]*lib/ocaml:"lib/ocaml:g' \
-    | sed -e 's:"/.*/\(\.opam\|opam2\)/[^/]*/\([^"]*\)":"~opam/\2":g' \
     | sed -e 's:\\n:\
 :g'
