diff -u liquidsoap.1.0.1.orig/configure.ac liquidsoap.1.0.1/configure.ac
--- liquidsoap.1.0.1.orig/configure.ac	2013-02-22 10:09:51.000000000 +0100
+++ liquidsoap.1.0.1/configure.ac	2013-02-22 10:10:41.000000000 +0100
@@ -246,14 +246,8 @@
 AC_MSG_CHECKING([for ocamlc version])
 AC_MSG_RESULT([$OCAML_MAJOR.$OCAML_MINOR.$OCAML_REVISION])
 
-# Variables for support of TCP_NODELAY:
-if test $OCAML_MAJOR -ge 3 -a $OCAML_MINOR -ge 11 ; then
-  NEED_TCP_NODELAY=
-  TCP_NODELAY_VALUE='let set_tcp_nodelay fd v = Unix.setsockopt fd Unix.TCP_NODELAY v'
-else
-  NEED_TCP_NODELAY=yes
-  TCP_NODELAY_VALUE='external set_tcp_nodelay : Unix.file_descr -> bool -> unit = "stub_set_tcp_nodelay"'
-fi
+NEED_TCP_NODELAY=
+TCP_NODELAY_VALUE='let set_tcp_nodelay fd v = Unix.setsockopt fd Unix.TCP_NODELAY v'
 AC_SUBST(NEED_TCP_NODELAY)
 AC_SUBST(TCP_NODELAY_VALUE)
 
Common subdirectories: liquidsoap.1.0.1.orig/doc and liquidsoap.1.0.1/doc
Common subdirectories: liquidsoap.1.0.1.orig/examples and liquidsoap.1.0.1/examples
Common subdirectories: liquidsoap.1.0.1.orig/gui and liquidsoap.1.0.1/gui
Common subdirectories: liquidsoap.1.0.1.orig/m4 and liquidsoap.1.0.1/m4
Common subdirectories: liquidsoap.1.0.1.orig/scripts and liquidsoap.1.0.1/scripts
Common subdirectories: liquidsoap.1.0.1.orig/src and liquidsoap.1.0.1/src
