diff -Naur camlidl-1.05/config/Makefile.unix camlidl-1.05-patched/config/Makefile.unix
--- camlidl-1.05/config/Makefile.unix	2002-04-22 12:50:46.000000000 +0100
+++ camlidl-1.05-patched/config/Makefile.unix	2016-05-04 13:10:23.370704022 +0100
@@ -37,7 +37,7 @@
 BINDIR=/usr/local/bin
 
 # The Objective Caml compilers (the defaults below should be OK)
-OCAMLC=ocamlc -g
+OCAMLC=ocamlc -g -warn-error -31
 OCAMLOPT=ocamlopt
 OCAMLYACC=ocamlyacc -v
 OCAMLLEX=ocamllex
diff -Naur camlidl-1.05/config/Makefile.win32 camlidl-1.05-patched/config/Makefile.win32
--- camlidl-1.05/config/Makefile.win32	2004-07-08 13:21:58.000000000 +0100
+++ camlidl-1.05-patched/config/Makefile.win32	2016-05-04 13:10:23.370704022 +0100
@@ -35,7 +35,7 @@
 BINDIR=C:/ocaml/bin
 
 # The Objective Caml compilers (the defaults below should be OK)
-OCAMLC=ocamlc -g
+OCAMLC=ocamlc -g -warn-error -31
 OCAMLOPT=ocamlopt
 OCAMLDEP=ocamldep
 OCAMLYACC=ocamlyacc -v
