diff -uNr javalib-2.3/src/jBasics.ml javalib-2.3a/src/jBasics.ml
--- javalib-2.3/src/jBasics.ml	2013-10-30 08:35:30.000000000 +0100
+++ javalib-2.3a/src/jBasics.ml	2015-06-18 15:36:17.000000000 +0200
@@ -278,8 +278,6 @@
       with _ ->
 	let cni = cnt.cni_next in
         let new_cn = (cni,cn) in
-	  if not (Str.string_match valid_class_name cn 0)
-	  then invalid_arg ("Error : " ^ cn ^ " is not a valid name for a class");
 	  cnt.cni_map <- ClassNameMap.add cn new_cn cnt.cni_map;
 	  cnt.cni_next <- cni + 1;
 	  new_cn
