diff --git a/_oasis b/_oasis
index f3733a5..4a7bec5 100644
--- a/_oasis
+++ b/_oasis
@@ -13,7 +13,7 @@ Description:
   MessagePack is an efficient binary serialization format.
   If meta_conv is installed, conv module will be installed.
 
-BuildTools: ocamlbuild, camlp4
+BuildTools: ocamlbuild
 
 # ------------------------------------------------------------
 # Flags
diff --git a/setup.ml b/setup.ml
index 26f0f36..2a51166 100644
--- a/setup.ml
+++ b/setup.ml
@@ -6991,8 +6991,7 @@ let setup_t =
                       bs_path = "lib/core";
                       bs_compiled_object = Best;
                       bs_build_depends = [FindlibPackage ("num", None)];
-                      bs_build_tools =
-                        [ExternalTool "ocamlbuild"; ExternalTool "camlp4"];
+                      bs_build_tools = [ExternalTool "ocamlbuild"];
                       bs_c_sources = [];
                       bs_data_files = [];
                       bs_ccopt = [(OASISExpr.EBool true, [])];
@@ -7054,8 +7053,7 @@ let setup_t =
                            FindlibPackage ("meta_conv", None);
                            InternalLibrary "msgpack"
                         ];
-                      bs_build_tools =
-                        [ExternalTool "ocamlbuild"; ExternalTool "camlp4"];
+                      bs_build_tools = [ExternalTool "ocamlbuild"];
                       bs_c_sources = [];
                       bs_data_files = [];
                       bs_ccopt = [(OASISExpr.EBool true, [])];
@@ -7102,8 +7100,7 @@ let setup_t =
                            InternalLibrary "msgpack";
                            FindlibPackage ("oUnit", None)
                         ];
-                      bs_build_tools =
-                        [ExternalTool "ocamlbuild"; ExternalTool "camlp4"];
+                      bs_build_tools = [ExternalTool "ocamlbuild"];
                       bs_c_sources = [];
                       bs_data_files = [];
                       bs_ccopt = [(OASISExpr.EBool true, [])];
@@ -7141,8 +7138,7 @@ let setup_t =
                            FindlibPackage ("oUnit", None);
                            FindlibPackage ("meta_conv.syntax", None)
                         ];
-                      bs_build_tools =
-                        [ExternalTool "ocamlbuild"; ExternalTool "camlp4"];
+                      bs_build_tools = [ExternalTool "ocamlbuild"];
                       bs_c_sources = [];
                       bs_data_files = [];
                       bs_ccopt = [(OASISExpr.EBool true, [])];
@@ -7185,7 +7181,6 @@ let setup_t =
                       test_tools =
                         [
                            ExternalTool "ocamlbuild";
-                           ExternalTool "camlp4";
                            InternalExecutable "test_core"
                         ]
                    });
@@ -7221,7 +7216,6 @@ let setup_t =
                       test_tools =
                         [
                            ExternalTool "ocamlbuild";
-                           ExternalTool "camlp4";
                            InternalExecutable "test_conv"
                         ]
                    });
