commit 942f9b16d8449d296e864830dbb40315033d4915
Author: Anil Madhavapeddy <anil@recoil.org>
Date:   Sun Jun 21 22:56:47 2020 +0100

    add `-fcommon` unconditionally to fix build on gcc10

diff --git a/configure b/configure
index dbc617807..9338088aa 100755
--- a/configure
+++ b/configure
@@ -425,6 +425,8 @@ case "$bytecc,$target" in
     bytecccompopts="-fno-defer-pop $gcc_warnings";;
 esac
 
+bytecccompopts="-fcommon $bytecccompopts"
+
 # Configure compiler to use in further tests
 
 cc="$bytecc -O $bytecclinkopts"
@@ -872,6 +874,7 @@ case "$arch,$nativecc,$system,$target" in
   *,gcc*,*,*)          nativecccompopts="$gcc_warnings";;
 esac
 
+nativecccompopts="-fcommon $nativecccompopts"
 asppprofflags='-DPROFILING'
 
 case "$arch,$system" in
