diff --git a/configure b/configure
index 10bce2c..6b25ac3 100755
--- a/configure
+++ b/configure
@@ -302,36 +302,6 @@ else
     exit 1
 fi >&2
 
-# - checking for java
-echo-check for java
-if which java &>/dev/null; then
-    echo-ok found
-else
-    echo-err Not found
-    exit 1
-fi
-
-# - checking for nodejs
-echo-check for nodejs
-if which node &>/dev/null; then
-    echo-ok found
-else
-    echo-err Not found
-    exit 1
-fi
-
-# - checking for nodejs version
-echo-check for nodejs version
-NODE_VERSION=$(node --version)
-NODE_MIN_VERSION="v0.6"
-if [[ $NODE_VERSION > $NODE_MIN_VERSION ]]; then
-    echo-ok $NODE_VERSION
-else
-    echo-err $NODE_VERSION
-    echo "Node $NODE_MIN_VERSION or newer is required"
-    exit 1
-fi
-
 # check C libs:
 #   we depend on libssl, libzip and others, but through
 #   ocaml libs, so no need to check them directly
