--- a/setup.ml	2019-04-16 15:02:32.357394398 +0100
+++ b/setup.ml	2019-04-16 15:02:45.789314884 +0100
@@ -7021,7 +7021,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3"
                              ]);
                            (OASISExpr.EAnd
@@ -7031,7 +7030,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-msse2";
                                 "-maes"
@@ -7197,7 +7195,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-DNDEBUG";
                                 "${XEN_CFLAGS}"
@@ -7209,7 +7206,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-DNDEBUG";
                                 "${XEN_CFLAGS}";
--- a/myocamlbuild.ml	2019-04-16 15:02:34.961379028 +0100
+++ b/myocamlbuild.ml	2019-04-16 16:15:30.751484060 +0100
@@ -654,8 +654,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3"
                    ]);
                (OASISExpr.EAnd
@@ -670,8 +668,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-msse2";
@@ -700,8 +696,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-DNDEBUG";
@@ -720,8 +714,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-DNDEBUG";
