--- lib/META.prev	2012-12-13 15:56:02.910434809 -0500
+++ lib/META	2012-12-13 15:56:39.167260697 -0500
@@ -3,7 +3,7 @@
 version = "109.07.00"
 description = "Jane Street Capital's standard library overlay"
 requires =
-"variantslib variantslib.syntax sexplib.syntax sexplib fieldslib.syntax fieldslib bin_prot bin_prot.syntax bigarray comparelib.syntax pa_ounit pa_pipebang res unix threads"
+"variantslib sexplib  fieldslib bin_prot  bigarray oUnit res unix threads"
 archive(byte) = "core.cma"
 archive(byte, plugin) = "core.cma"
 archive(native) = "core.cmxa"
