diff --git a/_oasis b/_oasis
index e7f0aab..8589fe9 100644
--- a/_oasis
+++ b/_oasis
@@ -27,16 +27,12 @@ Library xenopsd
   CompiledObject:     best
   Path:               lib
   Findlibname:        xenopsd
-  ByteOpt:            -warn-error +a
-  NativeOpt:          -warn-error +a
   Modules:            Suspend_image, Cancellable_subprocess, Bootloader, Ionice, Mac, Xenops_migrate, Xenops_hooks, Task_server, Xenops_task, Updates, Xenops_utils, Xenops_server, Xenopsd, Xenops_server_plugin, Xenops_server_skeleton, Scheduler, Path, Storage, Interface, Xenctrl_uuid, Xenstore, Version
   BuildDepends:       threads, threads.posix, uuidm, xmlm, cohttp, uri, rpclib, rpclib.syntax, forkexec, fd-send-recv, xcp, xcp.xen, xcp.storage, sexplib, sexplib.syntax, xcp, uutf, xenstore, xenstore.unix, xenstore_transport, xenstore_transport.unix
 
 Executable suspend_image_viewer
   CompiledObject:     best
   Path:               lib
-  ByteOpt:            -warn-error +a
-  NativeOpt:          -warn-error +a
   MainIs:             suspend_image_viewer.ml
   Install:            false
   BuildDepends:       xenopsd, cmdliner
@@ -44,8 +40,6 @@ Executable suspend_image_viewer
 Executable xenopsd_xc_main
   CompiledObject:     best
   Path:               xc
-  ByteOpt:            -warn-error +a
-  NativeOpt:          -warn-error +a
   MainIs:             xenops_xc_main.ml
   Build$:             flag(xen)
   Custom:             true
@@ -56,8 +50,6 @@ Executable xenopsd_xc_main
 Executable watch_test
   CompiledObject:     best
   Path:               xc
-  ByteOpt:            -warn-error +a
-  NativeOpt:          -warn-error +a
   MainIs:             watch_test.ml
   Install:            false
   BuildDepends:       xenopsd, xenstore, xenstore.unix, xenstore_transport, xenstore_transport.unix, threads
@@ -65,8 +57,6 @@ Executable watch_test
 Executable xenopsd_simulator
   CompiledObject:     best
   Path:               simulator
-  ByteOpt:            -warn-error +a
-  NativeOpt:          -warn-error +a
   MainIs:             xenops_simulator_main.ml
   Build$:             flag(simulator)
   Custom:             true
@@ -76,8 +66,6 @@ Executable xenopsd_simulator
 Executable xenopsd_libvirt_main
   CompiledObject:     best
   Path:               libvirt
-  ByteOpt:            -warn-error +a
-  NativeOpt:          -warn-error +a
   MainIs:             xenops_libvirt_main.ml
   Build$:             flag(libvirt)
   Custom:             true
@@ -87,8 +75,6 @@ Executable xenopsd_libvirt_main
 Executable xenopsd_xenlight_main
   CompiledObject:     best
   Path:               xl
-  ByteOpt:            -warn-error +a
-  NativeOpt:          -warn-error +a
   MainIs:             xenops_xl_main.ml
   Build$:             flag(xenlight)
   Custom:             true
diff --git a/configure b/configure
index 5ad76e3..8540c77 100755
--- a/configure
+++ b/configure
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+oasis setup
+
 D=$(mktemp -d /tmp/configure.XXXXX)
 function cleanup {
   cd /
