--- Makefile-	2015-12-14 16:15:26.261118345 +0100
+++ Makefile	2015-12-14 16:15:32.093254187 +0100
@@ -21,16 +21,13 @@
 # | 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA    |
 # +-----------------------------------------------------------------+
 
-# path to the MagickCore-config utility
-MAGICK_INSTALLED_BIN := $(shell which MagickCore-config)
+MAGICK_INSTALLED_BIN := pkg-config
 
-MAGICK_PREFIX := $(shell $(MAGICK_INSTALLED_BIN) --prefix)
-
-MAGICK_CLIBS := $(shell $(MAGICK_INSTALLED_BIN) --libs)
+MAGICK_CLIBS := $(shell $(MAGICK_INSTALLED_BIN) --libs ImageMagick)
 
 MAGICK_CLIBS_ := $(shell ocaml mlarg.ml $(MAGICK_CLIBS))
 
-MAGICK_CFLAGS := $(shell $(MAGICK_INSTALLED_BIN) --cflags)
+MAGICK_CFLAGS := $(shell $(MAGICK_INSTALLED_BIN) --cflags ImageMagick)
 
 OCAML_DIR := $(shell ocamlc -where)
 #OCAML_DIR := $(shell ocamlfind printconf stdlib)
