diff --git a/algo/dominators.ml b/algo/dominators.ml
index d24ae09..8681b96 100644
--- a/algo/dominators.ml
+++ b/algo/dominators.ml
@@ -99,7 +99,7 @@ let dominators_tarjan graph =
   ) graph;
 
   Util.Timer.start tjntimer;
-  let module Dom = Dominator.Make(G) in
+  let module Dom = Dominator.Make_graph(G) in
   let idom = Dom.compute_all graph start_pkg in
   let domgr = idom.Dom.dom_graph () in
   Util.Timer.stop tjntimer ();
