--- camlspotter-planck-fbb6d4646a91.orig/ocaml/OMakefile	2013-07-09 02:53:08.000000000 -0400
+++ camlspotter-planck-fbb6d4646a91/ocaml/OMakefile	2015-03-24 13:17:13.176560897 -0400
@@ -33,6 +33,7 @@
 
 NO_INSTALL=true
 MyOCamlProgram(plpautoconv, $(FILES_plpautoconv))
+plpautoconv.cmi plpautoconv.cmx plpautoconv.o plpautoconv.cmo: ../pa_bind_inline/pa_bind_inline.cmo
 
 ################################################################# lexer test
 
