--- a/ukvm/ukvm-core.c
+++ b/ukvm/ukvm-core.c
@@ -36,7 +36,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <linux/const.h>
-#include <asm/msr-index.h>
 #include <elf.h>
 #include <errno.h>
 #include <assert.h>
--- a/ukvm/ukvm-cpu.h
+++ b/ukvm/ukvm-cpu.h
@@ -133,4 +133,7 @@ struct _kvm_segment {
         seg.avl = GDT_GET_AVL(gdt_ent);         \
     } while (0)
 
+#define EFER_LME_BIT            8 /* Long mode enable (R/W) */
+#define EFER_LME                _BITUL(EFER_LME_BIT)
+
 #endif
