diff --git a/filesys-1.1/static/notebook/js/notebook.js b/filesys-1.1/static/notebook/js/notebook.js
index 488364b..181e6ec 100644
--- a/filesys-1.1/static/notebook/js/notebook.js
+++ b/filesys-1.1/static/notebook/js/notebook.js
@@ -321,7 +321,7 @@ var IPython = (function (IPython) {
                 that.split_cell();
                 that.control_key_active = false;
                 return false;                                               
-            } else if (that.control_key_active) {
+            } else if (event.which !== 229 && that.control_key_active) { /* work around a chromium bug */
                 that.control_key_active = false;
                 return true;
             }
