--- Makefile.in.orig	2013-02-14 13:58:30.984219962 +0000
+++ Makefile.in	2013-02-14 13:58:40.380219678 +0000
@@ -64,8 +64,8 @@
 CAMLP4   = @CAMLP4O@
 
 INCLUDES = -I src -I jc -I c -I java -I intf -I tools -I mix -I ml
-BFLAGS   = -w Z -warn-error A -dtypes -g $(INCLUDES) @INCLUDEGTK2@ -I +threads @OCAMLGRAPHLIB@
-OFLAGS   = -w Z -warn-error A -dtypes $(INCLUDES) @INCLUDEGTK2@ -I +threads @OCAMLGRAPHLIB@
+BFLAGS   = -w Z -dtypes -g $(INCLUDES) @INCLUDEGTK2@ -I +threads @OCAMLGRAPHLIB@
+OFLAGS   = -w Z -dtypes $(INCLUDES) @INCLUDEGTK2@ -I +threads @OCAMLGRAPHLIB@
 
 LCFLAGS = -L/usr/lib -L/usr/local/lib/ocaml
 
