--- a/lib/coq/WhyTuples.v
+++ b/lib/coq/WhyTuples.v
@@ -23,8 +23,8 @@ Definition tuple_1 (X:Set) := X.
 Definition tuple_2 := prod.
 Definition Build_tuple_2 (A:Set) (B:Set) := pair A B.
 Implicit Arguments Build_tuple_2.
-Definition proj_2_1 := fst.
-Definition proj_2_2 := snd.
+Definition proj_2_1 (X:Set * Set) := fst X.
+Definition proj_2_2 (X:Set * Set) := snd X.
 
 Record tuple_3 (T1 T2 T3:Set) : Set := 
   {proj_3_1 : T1; proj_3_2 : T2; proj_3_3 : T3}.
