--- a/ukvm/ukvm-configure
+++ b/ukvm/ukvm-configure
@@ -40,7 +40,7 @@
 
 UKVM_CC?=cc
 UKVM_FLAGS=-D__UKVM_HOST__ \$(UKVM_MODULE_FLAGS)
-UKVM_CFLAGS=-Wall -Werror -std=c99 -O2 -g \$(UKVM_FLAGS)
+UKVM_CFLAGS=-Wall -std=c99 -O2 -g \$(UKVM_FLAGS)
 UKVM_OBJS=_build-ukvm/ukvm-core.o \$(UKVM_MODULE_OBJS)
 ifdef UKVM_STATIC
 UKVM_LDFLAGS=-static
--- a/Makefile.common
+++ b/Makefile.common
@@ -32,4 +32,4 @@
 # Likewise.
 LDFLAGS=-nostdlib -z max-page-size=0x1000 -static
 # CFLAGS used for building kernel/ and in-tree tests.
-CFLAGS=$(MD_CFLAGS) -isystem $(TOP)/include-host -std=gnu99 -Wall -Wextra -Werror -O2 -g
+CFLAGS=$(MD_CFLAGS) -isystem $(TOP)/include-host -std=gnu99 -Wall -Wextra -O2 -g
