commit 7d43a4bca5c11668d4a96872c96eb6af1d9fb253
Author: Kate <kit.ty.kate@disroot.org>
Date:   Wed Apr 10 13:49:26 2019 +0100

    Fix Makefile generation for the OCaml api

diff --git a/scripts/mk_util.py b/scripts/mk_util.py
index b8fbf9a45..3b71af793 100644
--- a/scripts/mk_util.py
+++ b/scripts/mk_util.py
@@ -2255,7 +2255,7 @@ class MLComponent(Component):
             for m in self.modules:
                 ff = os.path.join(src_dir, m + '.ml')
                 ft = os.path.join(self.sub_dir, m + '.cmx')
-                out.write('%s: %s %s\n' % (ft, ff, cmos))
+                out.write('%s: %s %s %s\n' % (ft, ff, cmos, cmxs))
                 out.write('\t%s -I %s -o %s -c %s\n' % (OCAMLOPTF, self.sub_dir, ft, ff))
                 cmxs = cmxs + ' ' + ft
 
