diff -r -u obigstore-obigstore-0.9.1.orig/OMakefile obigstore-obigstore-0.9.1/OMakefile
--- obigstore-obigstore-0.9.1.orig/OMakefile	2014-01-08 08:00:20.000000000 -0500
+++ obigstore-obigstore-0.9.1/OMakefile	2015-04-07 14:25:59.056019978 -0400
@@ -31,7 +31,7 @@
 .PHONY: install
 .PHONY: uninstall
 
-OCAMLFLAGS     = -g -thread -w +a-4-6-9-27..29-32..99 -warn-error +a-4-6-9-18-27-28-29..99
+OCAMLFLAGS     = -g -thread -w +a-4-6-9-27..29-32..99 -warn-error -a
 OCAMLFINDFLAGS = -syntax camlp4o -ppopt -lwt-debug
 CFLAGS         = -g -Wall -O2 -fPIC
 
