From f6ac7490cf262b4669081282feab7bccbea22e1a Mon Sep 17 00:00:00 2001
From: OCamlPro-Iguernlala <mohamed.iguernlala@ocamlpro.com>
Date: Thu, 26 Apr 2018 19:09:41 +0200
Subject: [PATCH] fix uninstall-lib in Makefile

---
 sources/Makefile.users | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sources/Makefile.users b/sources/Makefile.users
index fdad643..16d2be4 100644
--- a/Makefile.users
+++ b/Makefile.users
@@ -263,7 +263,7 @@ install-lib: lib META
 #	cp -f META $(LIBDIR)
 
 uninstall-lib:
-	ocamlfind remove alt-ergo-lib
+	ocamlfind remove alt-ergo
 
 install: install-ae install-gui install-fm-simplex install-lib
 
2.14.1

