diff --git a/src/plugins/value/gen-api.sh b/src/plugins/value/gen-api.sh
index d00c16fdd4a..a64e6268c2d 100755
--- a/src/plugins/value/gen-api.sh
+++ b/src/plugins/value/gen-api.sh
@@ -22,7 +22,8 @@ for i in "$@"
 do
     file=$(basename $i)
     module=${file%.*}
-    printf '\nmodule %s: sig\n' ${module^}
+    Module="$(echo "${module:0:1}" | tr '[:lower:]' '[:upper:]')${module:1}"
+    printf '\nmodule %s: sig\n' $Module
     awk '/\[@@@ api_start\]/{flag=1;next} /\[@@@ api_end\]/{flag=0} flag{ print (NF ? "  ":"") $0 }' $i
     printf 'end\n'
 done
