diff -Naur lwt-4.2.1.orig/lwt.opam lwt-4.2.1/lwt.opam
--- lwt-4.2.1.orig/lwt.opam	2020-12-30 22:06:53.640469206 +0100
+++ lwt-4.2.1/lwt.opam	2020-12-30 22:08:52.634458990 +0100
@@ -2,7 +2,7 @@
 
 synopsis: "Promises and event-driven I/O"
 
-version: "dev"
+version: "4.2.1"
 license: "MIT"
 homepage: "https://github.com/ocsigen/lwt"
 doc: "https://ocsigen.org/lwt/manual/"
diff -Naur lwt-4.2.1.orig/lwt_ppx.opam lwt-4.2.1/lwt_ppx.opam
--- lwt-4.2.1.orig/lwt_ppx.opam	2020-12-30 22:06:53.640469206 +0100
+++ lwt-4.2.1/lwt_ppx.opam	2020-12-30 22:08:57.079458608 +0100
@@ -2,7 +2,7 @@
 
 synopsis: "PPX syntax for Lwt, providing something similar to async/await from JavaScript"
 
-version: "dev"
+version: "4.2.1"
 license: "MIT"
 homepage: "https://github.com/ocsigen/lwt"
 doc: "https://ocsigen.org/lwt/api/Ppx_lwt"
diff -Naur lwt-4.2.1.orig/lwt_react.opam lwt-4.2.1/lwt_react.opam
--- lwt-4.2.1.orig/lwt_react.opam	2020-12-30 22:06:53.640469206 +0100
+++ lwt-4.2.1/lwt_react.opam	2020-12-30 22:09:00.320458330 +0100
@@ -2,7 +2,7 @@
 
 synopsis: "Helpers for using React with Lwt"
 
-version: "dev"
+version: "4.2.1"
 license: "MIT"
 homepage: "https://github.com/ocsigen/lwt"
 doc: "https://ocsigen.org/lwt/api/Lwt_react"
