diff --git a/Makefile b/Makefile
index b5f718c..3487c4a 100644
--- a/Makefile
+++ b/Makefile
@@ -7,8 +7,8 @@ RESULT = cpdf
 ANNOTATE = true
 PACKS = camlpdf
 
-OCAMLNCFLAGS = -g -w -3 -annot
-OCAMLBCFLAGS = -g -w -3 -annot
+OCAMLNCFLAGS = -g -w -3 -annot -unsafe-string
+OCAMLBCFLAGS = -g -w -3 -annot -unsafe-string
 OCAMLLDFLAGS = -g
 
 all : native-code native-code-library byte-code-library top htdoc
