--- a/_oasis	2019-04-14 18:53:58.732983307 +0100
+++ b/_oasis	2019-04-16 13:22:25.459958829 +0100
@@ -73,7 +73,7 @@
   XMETARequires:   cstruct, zarith, sexplib
   XMETAExtraLines: xen_linkopts = "-lnocrypto_xen_stubs"
   ByteOpt:         -w A-4-29-33-40-41-42-43-34-44
-  CCOpt:           --std=c99 -Wall -Wextra -Werror -O3
+  CCOpt:           --std=c99 -Wall -Wextra -O3
   if flag(modernity) && architecture(amd64)
     CCOpt+: -DACCELERATE -msse2 -maes
 
@@ -117,7 +117,7 @@
                    native/des/generic.h
   BuildDepends:    nocrypto, mirage-entropy-xen
   XMETARequires:   nocrypto, lwt, mirage-entropy-xen
-  CCOpt:           --std=c99 -Wall -Wextra -Werror -O3 -DNDEBUG $XEN_CFLAGS
+  CCOpt:           --std=c99 -Wall -Wextra -O3 -DNDEBUG $XEN_CFLAGS
   if flag(modernity) && architecture(amd64)
     CCOpt+: -DACCELERATE -msse2 -maes
 
