--- coqide.8.4pl2/configure	2012-12-22 11:06:16.000000000 +0100
+++ coqide.8.4pl2/configure	2013-01-18 16:30:02.110785838 +0100
@@ -887,7 +887,7 @@
     */true/*/*) COQRUNBYTEFLAGS="-dllib -lcoqrun -dllpath '$COQTOP'/kernel/byterun";;
     *)
         COQRUNBYTEFLAGS="-dllib -lcoqrun -dllpath '$LIBDIR'"
-        BUILDLDPATH="export CAML_LD_LIBRARY_PATH='$COQTOP'/kernel/byterun";;
+        BUILDLDPATH="export CAML_LD_LIBRARY_PATH='$COQTOP'/kernel/byterun:$CAML_LD_LIBRARY_PATH";;
 esac
 case $coqtoolsbyteflags_spec/$custom_spec/$CUSTOM_OS in
     yes/*/*) COQTOOLSBYTEFLAGS="$coqtoolsbyteflags";;
