diff -ur arakoon-1.6.5/Makefile arakoon-1.6.5-opam/Makefile
--- arakoon-1.6.5/Makefile	2013-10-01 17:45:58.000000000 +0200
+++ arakoon-1.6.5-opam/Makefile	2013-12-10 13:22:02.323993019 +0100
@@ -1,7 +1,7 @@
 # This makefile wrapper makes DEBIAN packaging easier. 
 
 OCAML_VERSION ?= 4.00.1
-OCAML_LIBDIR ?= $(DESTDIR)/usr/lib/ocaml/
+OCAML_LIBDIR ?= $(DESTDIR)/lib/
 OCAML_FIND ?= ocamlfind
 
 all: build
@@ -18,8 +18,8 @@
 install: install_client install_server
 
 install_server:
-	mkdir -p $(DESTDIR)/usr/bin/
-	cp ./arakoon.native $(DESTDIR)/usr/bin/arakoon
+	mkdir -p $(DESTDIR)/bin/
+	cp ./arakoon.native $(DESTDIR)/bin/arakoon
 
 install_client:
 	mkdir -p $(OCAML_LIBDIR)
