diff --git a/lib_test/test_lib.ml b/lib_test/test_lib.ml
index 6364eff..c878845 100644
--- a/lib_test/test_lib.ml
+++ b/lib_test/test_lib.ml
@@ -88,7 +88,7 @@ let run name suites =
   let num_tests = List.fold suites ~init:0 ~f:(fun sum suite ->
     sum + List.length suite.tests)
   in
-  Printf.printf "Running %d tests for library %s.\n%!" num_tests name;
+  printf "Running %d tests for library %s.\n%!" num_tests name;
   let rec loop_suites suites counter failures =
     match suites with
     | [] ->
