diff --git a/ocamlutil/pretty.ml b/ocamlutil/pretty.ml
index 9f2fc43..4118097 100644
--- a/ocamlutil/pretty.ml
+++ b/ocamlutil/pretty.ml
@@ -640,7 +640,7 @@ external format_float: string -> float -> string = "caml_format_float"
     
 let gprintf (finish : doc -> 'b)  
             (format : ('a, unit, doc, 'b) format4) : 'a =
-  let format = (Obj.magic format : string) in
+  let format = string_of_format format in
 
   (* Record the starting align depth *)
   let startAlignDepth = !alignDepth in
