diff -ru pgocaml.1.6/Makefile pgocaml.1.6/Makefile
--- pgocaml.1.6/Makefile	2012-07-04 19:53:31.000000000 +0200
+++ pgocaml.1.6/Makefile	2012-10-05 18:14:39.889285053 +0200
@@ -166,7 +166,7 @@
 #
 
 findlib_install:
-	ocamlfind install $(PACKAGE) META pgocaml.a pgocaml.cma pgocaml.cmxa pGOCaml_generic.cm[ix] pGOCaml.cm[ix] pa_pgsql.cmo pGOCaml_generic.mli pGOCaml.mli
+	ocamlfind install $(PACKAGE) META pgocaml.cmxs pgocaml.a pgocaml.cma pgocaml.cmxa pGOCaml_generic.cm[ix] pGOCaml.cm[ix] pa_pgsql.cmo pGOCaml_generic.mli pGOCaml.mli
 
 reinstall:
 	ocamlfind remove $(PACKAGE)
