diff --git a/apron/Makefile b/apron/Makefile
index 8be6271..50fbe8e 100644
--- a/apron/Makefile
+++ b/apron/Makefile
@@ -5,7 +5,7 @@ LCFLAGS = \
 -L$(PPL_PREFIX)/lib -Lppl \
 -Lproducts \
 -L$(GMP_PREFIX)/lib -L$(MPFR_PREFIX)/lib \
--L$(CAMLIDL_PREFIX)/lib/ocaml
+-L$(CAMLIDL_PREFIX)/lib/camlidl
 
 all: c
 ifneq ($(HAS_OCAML),)
diff --git a/apron/box/Makefile b/apron/box/Makefile
index 2dc4cfa..ea52f18 100644
--- a/apron/box/Makefile
+++ b/apron/box/Makefile
@@ -28,7 +28,7 @@ ICFLAGS = \
 -I../apron \
 -I../mlapronidl \
 -I$(GMP_PREFIX)/include -I$(MPFR_PREFIX)/include \
--I$(CAMLIDL_PREFIX)/lib/ocaml -I$(CAML_PREFIX)/lib/ocaml
+-I$(CAMLIDL_PREFIX)/lib/camlidl -I$(CAML_PREFIX)/lib/ocaml
 
 # Caml
 OCAMLINC = -I $(MLGMPIDL_LIB) -I ../mlapronidl
diff --git a/apron/examples/Makefile b/apron/examples/Makefile
index 2735c67..43b3a3b 100644
--- a/apron/examples/Makefile
+++ b/apron/examples/Makefile
@@ -13,7 +13,7 @@ LCFLAGS = \
 -L$(MPFR_PREFIX)/lib \
 -L$(APRON_PREFIX)/lib \
 -L$(PPL_PREFIX)/lib \
--L$(CAMLIDL_PREFIX)/lib/ocaml
+-L$(CAMLIDL_PREFIX)/lib/camlidl
 
 OCAMLINC = \
 -I $(MLGMPIDL_PREFIX)/lib \
diff --git a/apron/mlapronidl/Makefile b/apron/mlapronidl/Makefile
index cb6848a..727a6bc 100644
--- a/apron/mlapronidl/Makefile
+++ b/apron/mlapronidl/Makefile
@@ -30,7 +30,7 @@ ICFLAGS = \
 -I$(MLGMPIDL_INCLUDE) -I../apron \
 -I$(GMP_PREFIX)/include \
 -I$(MPFR_PREFIX)/include \
--I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/ocaml
+-I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/camlidl
 
 #---------------------------------------
 # Files
diff --git a/apron/newpolka/Makefile b/apron/newpolka/Makefile
index 709a345..9ca8249 100644
--- a/apron/newpolka/Makefile
+++ b/apron/newpolka/Makefile
@@ -28,7 +28,7 @@ ICFLAGS = \
 -I../itv \
 -I../mlapronidl \
 -I$(GMP_PREFIX)/include -I$(MPFR_PREFIX)/include \
--I$(CAMLIDL_PREFIX)/lib/ocaml -I$(CAML_PREFIX)/lib/ocaml
+-I$(CAMLIDL_PREFIX)/lib/camlidl -I$(CAML_PREFIX)/lib/ocaml
 
 # Caml
 OCAMLINC = -I $(MLGMPIDL_LIB) -I ../mlapronidl
diff --git a/apron/octagons/Makefile b/apron/octagons/Makefile
index 015616b..29a08bf 100644
--- a/apron/octagons/Makefile
+++ b/apron/octagons/Makefile
@@ -43,7 +43,7 @@ ICFLAGS = \
 -I../num \
 -I../itv \
 -I$(MPFR_PREFIX)/include -I$(GMP_PREFIX)/include \
--I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/ocaml
+-I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/camlidl
 
 # Caml
 OCAMLINC = -I $(MLGMPIDL_LIB) -I ../mlapronidl
diff --git a/apron/ppl/Makefile b/apron/ppl/Makefile
index 3cb2d61..cb67380 100644
--- a/apron/ppl/Makefile
+++ b/apron/ppl/Makefile
@@ -45,7 +45,7 @@ ICFLAGS = \
 -I../itv \
 -I$(GMP_PREFIX)/include -I$(MPFR_PREFIX)/include \
 -I$(PPL_PREFIX)/include \
--I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/ocaml
+-I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/camlidl
 
 ICXXFLAGS = \
 -I../newpolka \
@@ -56,7 +56,7 @@ ICXXFLAGS = \
 -I../itv \
 -I$(GMP_PREFIX)/include -I$(MPFR_PREFIX)/include \
 -I$(PPL_PREFIX)/include \
--I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/ocaml
+-I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/camlidl
 
 # Caml
 OCAMLINC = -I $(MLGMPIDL_LIB) -I ../mlapronidl
diff --git a/apron/products/Makefile b/apron/products/Makefile
index b802dd4..7c71d3a 100644
--- a/apron/products/Makefile
+++ b/apron/products/Makefile
@@ -25,7 +25,7 @@ ICFLAGS = \
 -I../ppl \
 -I../mlapronidl \
 -I$(GMP_PREFIX)/include -I$(MPFR_PREFIX)/include \
--I$(CAMLIDL_PREFIX)/lib/ocaml -I$(CAML_PREFIX)/lib/ocaml
+-I$(CAMLIDL_PREFIX)/lib/camlidl -I$(CAML_PREFIX)/lib/ocaml
 
 # Caml
 OCAMLINC = -I $(MLGMPIDL_LIB) -I ../mlapronidl -I ../newpolka -I ../ppl
diff --git a/apron/taylor1plus/Makefile b/apron/taylor1plus/Makefile
index 1829f6c..2558ecb 100644
--- a/apron/taylor1plus/Makefile
+++ b/apron/taylor1plus/Makefile
@@ -37,7 +37,7 @@ ICFLAGS = \
 -I../apron \
 -I../mlapronidl \
 -I$(GMP_PREFIX)/include -I$(MPFR_PREFIX)/include \
--I$(CAMLIDL_PREFIX)/lib/ocaml -I$(CAML_PREFIX)/lib/ocaml
+-I$(CAMLIDL_PREFIX)/lib/camlidl -I$(CAML_PREFIX)/lib/ocaml
 
 # Caml
 OCAMLINC = -I $(MLGMPIDL_LIB) -I ../mlapronidl
diff --git a/apron/test/Makefile b/apron/test/Makefile
index 02c8fa4..124440b 100644
--- a/apron/test/Makefile
+++ b/apron/test/Makefile
@@ -11,7 +11,7 @@ LCFLAGS = \
 -L$(MPFR_PREFIX)/lib \
 -L$(APRON_PREFIX)/lib \
 -L$(PPL_PREFIX)/lib \
--L$(CAMLIDL_PREFIX)/lib/ocaml
+-L$(CAMLIDL_PREFIX)/lib/camlidl
 
 OCAMLINC = \
 -I $(MLGMPIDL_PREFIX)/lib \
diff --git a/mlgmpidl/Makefile b/mlgmpidl/Makefile
index db542de..bf262da 100644
--- a/mlgmpidl/Makefile
+++ b/mlgmpidl/Makefile
@@ -28,7 +28,7 @@ endif
 
 
 OCAMLCCOPT += \
--ccopt -L$(CAMLIDL_PREFIX)/lib/ocaml \
+-ccopt -L$(CAMLIDL_PREFIX)/lib/camlidl \
 -ccopt -L$(CAML_PREFIX)/lib/ocaml \
 -ccopt -L$(GMP_PREFIX)/lib \
 -ccopt -L$(MPFR_PREFIX)/lib
@@ -43,12 +43,12 @@ MLMODULES = $(IDLMODULES) mpzf mpqf mpfrf
 ICFLAGS = \
 -I$(GMP_PREFIX)/include \
 -I$(MPFR_PREFIX)/include \
--I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/ocaml
+-I$(CAML_PREFIX)/lib/ocaml -I$(CAMLIDL_PREFIX)/lib/camlidl
 
 LDFLAGS = \
 -L$(GMP_PREFIX)/lib \
 -L$(MPFR_PREFIX)/lib \
--L$(CAML_PREFIX)/lib/ocaml -L$(CAML_PREFIX)/lib/ocaml/stublibs -L$(CAMLIDL_PREFIX)/lib/ocaml
+-L$(CAML_PREFIX)/lib/ocaml -L$(CAML_PREFIX)/lib/ocaml/stublibs -L$(CAMLIDL_PREFIX)/lib/camlidl
 
 CCMODULES = $(IDLMODULES:%=%_caml) gmp_caml
 
