diff --git a/src/Makefile b/src/Makefile
index 22b1928..cd7c283 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -36,12 +36,12 @@ topdir = ..
 OTTVER=0.26
 
 # for us
-OCAMLC	 	= ocamlc.opt -g -dtypes  # -w p disables partial match warnings
-OCAMLOPT	= ocamlopt.opt -w y -g -dtypes -unsafe -inline 9
+#OCAMLC	 	= ocamlc.opt -g -dtypes  # -w p disables partial match warnings
+#OCAMLOPT	= ocamlopt.opt -w y -g -dtypes -unsafe -inline 9
 
 # for releases
-#OCAMLC	 	= ocamlc -g -w p -w y
-#OCAMLOPT	= ocamlopt -w p -w y -unsafe -inline 9
+OCAMLC	 	= ocamlc -g -w p -w y
+OCAMLOPT	= ocamlopt -w p -w y -unsafe -inline 9
 
 OCAMLDEP 	= ocamldep
 OCAMLLEX        = ocamllex
@@ -105,8 +105,8 @@ CMX 	= $(patsubst %.cmo, %.cmx, $(CMO))
 CMI_FROM_MLI = $(patsubst %.mli, %.cmi, $(SOURCES_MLI))
 
 
-#default: ott.opt
-default: ott.byt
+default: ott.opt
+#default: ott.byt
 
 all: default
 ott: default
