--- ocaml-bitstring-2.0.4.orig/Makefile.in	2016-08-12 14:47:34.000000000 +0200
+++ ocaml-bitstring-2.0.4/Makefile.in	2016-08-12 14:50:51.000000000 +0200
@@ -110,7 +110,7 @@
 	  -I +camlp4 -pp camlp4of -c $<
 
 pa_bitstring.cmo: pa_bitstring.ml bitstring.cma bitstring_persistent.cma
-	$(OCAMLFIND) ocamlc bitstring.cma -I +camlp4 dynlink.cma camlp4lib.cma \
+	$(OCAMLFIND) ocamlc -I +camlp4 \
 	  -pp camlp4of -c $< -o $@
 
