--- a/flatestubs.c	2016-03-13 20:03:44.000000000 +0000
+++ b/flatestubs.c	2016-03-13 20:02:13.000000000 +0000
@@ -19,6 +19,8 @@
 
 #include "miniz.c" //Modified to use miniz.c rather than zlib 
 
+#include <stdint.h>
+
 #include <caml/mlvalues.h>
 #include <caml/alloc.h>
 #include <caml/callback.h>
@@ -170,7 +172,7 @@
 
 value camlzip_update_crc32(value crc, value buf, value pos, value len)
 {
-  return copy_int32(crc32((uint32) Int32_val(crc), 
+  return copy_int32(crc32((uint32_t) Int32_val(crc), 
                           &Byte_u(buf, Long_val(pos)),
                           Long_val(len)));
 }
