diff -ru cairo.0.4.1/setup.ml cairo.0.4.1/setup.ml
--- cairo.0.4.1/setup.ml	2011-12-30 17:23:59.000000000 +0100
+++ cairo.0.4.1/setup.ml	2012-10-05 18:15:15.442359704 +0200
@@ -5999,7 +5999,7 @@
       | "Unix" | "Cygwin" -> "-o " ^ exe
       | "Win32" -> "/Fe" ^ exe
       | _ -> assert false in
-    let args = o :: stdlib :: (flags @ [tmp]) in
+    let args = o :: stdlib :: (tmp :: flags @ []) in
     let f_exit_code e =
       if e <> 0 then (compile_err(); Sys.remove tmp; exit 1) in
     BaseExec.run cc args ~f_exit_code;
