diff -ru missinglib.0.4.1/Makefile.setup missinglib.0.4.1/Makefile.setup
--- missinglib.0.4.1/Makefile.setup	2004-06-05 17:45:01.000000000 +0200
+++ missinglib.0.4.1/Makefile.setup	2012-10-05 18:14:52.729673166 +0200
@@ -16,7 +16,7 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
-OCAMLFLAGS := -warn-error A
+OCAMLFLAGS := -w -a
 
 ifndef OCAMLMAKEFILE
   OCAMLMAKEFILE := $(PATHTOTOP)/contrib/OCamlMakefile/OCamlMakefile
