--- gmp-6.0.0.orig.orig/config.guess	2014-03-25 14:37:55.000000000 +0000
+++ gmp-6.0.0.orig/config.guess	2014-09-16 15:46:57.457485087 +0000
@@ -173,7 +173,7 @@
   ;;
 
 arm*-*-*)
-  cpu_code=`sed -n 's/^CPU part.*\(0x.*\)$/\1/p' /proc/cpuinfo 2>/dev/null`
+  cpu_code=`sed -n 's/^CPU part.*\(0x.*\)$/\1/p' /proc/cpuinfo | head -1 2>/dev/null`
   case "$cpu_code" in
       0xa10 | 0xa11 | 0xb11)			# v4 strongarm/sa1100
 		exact_cpu="armsa1";;
@@ -200,7 +200,7 @@
       0xc15)	exact_cpu="armcortexr5";;	# v7r
       0xc23)	exact_cpu="armcortexm3";;	# v7m
   esac
-  exact_cpu="${exact_cpu}`sed -n 's;^Features.*\(neon\).*;\1;p' /proc/cpuinfo 2>/dev/null`"
+  exact_cpu="${exact_cpu}`sed -n 's;^Features.*\(neon\).*;\1;p' /proc/cpuinfo | head -1 2>/dev/null`"
   ;;
 
 ia64*-*-*)
