--- a/setup.ml	2019-04-16 15:03:37.793002055 +0100
+++ b/setup.ml	2019-04-16 15:03:49.008933634 +0100
@@ -7020,7 +7020,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3"
                              ]);
                            (OASISExpr.EAnd
@@ -7030,7 +7029,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-msse2";
                                 "-maes"
@@ -7196,7 +7194,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-DNDEBUG";
                                 "${XEN_CFLAGS}"
@@ -7208,7 +7205,6 @@
                                 "--std=c99";
                                 "-Wall";
                                 "-Wextra";
-                                "-Werror";
                                 "-O3";
                                 "-DNDEBUG";
                                 "${XEN_CFLAGS}";
--- a/myocamlbuild.ml	2019-04-16 15:03:35.797014198 +0100
+++ b/myocamlbuild.ml	2019-04-16 15:04:00.636862379 +0100
@@ -654,8 +654,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3"
                    ]);
                (OASISExpr.EAnd
@@ -670,8 +668,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-msse2";
@@ -700,8 +696,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-DNDEBUG";
@@ -720,8 +714,6 @@
                       A "-ccopt";
                       A "-Wextra";
                       A "-ccopt";
-                      A "-Werror";
-                      A "-ccopt";
                       A "-O3";
                       A "-ccopt";
                       A "-DNDEBUG";
