From 82eb7ee561c3e660dc33a6938346abf7354a23d4 Mon Sep 17 00:00:00 2001
From: David Allsopp <david.allsopp@metastack.com>
Date: Sat, 16 Dec 2017 11:05:30 +0000
Subject: [PATCH] Improve documentation installation with opam

---
 Makefile      | 10 +++++++++-
 bitmasks.opam |  5 +----
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index ee30633..c7e317a 100644
--- a/Makefile
+++ b/Makefile
@@ -31,8 +31,16 @@ JBUILDER=jbuilder
 build:
 	$(JBUILDER) build @install
 
-doc:
+bitmasks.install:
+	$(JBUILDER) build $@
+
+doc: bitmasks.install
 	$(JBUILDER) build @doc
+	@sed -e '$$d' $< | fgrep -v _doc > $<.tmp
+	@find _build/default/_doc -type f -not -name .jbuilder-keep | \
+    sed -e 's/.*/  "\0"/' -e 's/_doc\/\(.*\/.*\)/\0 {"\1}/' >> $<.tmp
+	@echo ']' >> $<.tmp
+	@mv $<.tmp $<
 
 test:
 	$(JBUILDER) runtest
diff --git a/bitmasks.opam b/bitmasks.opam
index bbd401d..ec4b7a4 100644
--- a/bitmasks.opam
+++ b/bitmasks.opam
@@ -7,10 +7,7 @@ homepage: "https://metastack.github.io/bitmasks"
 dev-repo: "https://github.com/metastack/bitmasks.git"
 bug-reports: "https://github.com/metastack/bitmasks/issues"
 build: [ "jbuilder" "build" "@install" ]
-build-doc: [
-  ["jbuilder" "build" "@doc"]
-  ["cp" "-r" "_build/default/_doc/bitmasks" "_build/default/_doc/index.html" "_build/default/_doc/odoc.css" "%{doc}%/"]
-]
+build-doc: [make "doc"]
 build-test: ["jbuilder" "runtest"]
 depends: [
   "jbuilder" {build}
