diff --git a/depends.ocp.in b/depends.ocp.in
index 7c3e312..6d284ee 100644
--- a/depends.ocp.in
+++ b/depends.ocp.in
@@ -32,13 +32,13 @@ begin library "arg"
 end
 
 begin library "cudf"
-  dirname  += "%{lib}%/mancoosi-cudf"
+  dirname  += "%{lib}%/cudf"
   generated = true
   requires  = [ "extLib" ]
 end
 
 begin library "dose"
-  dirname  += "%{lib}%/mancoosi-dose"
+  dirname  += "%{lib}%/dose"
   generated = true
   requires  = [ "re_pcre" "extLib" "cudf" "graph" "unix" "str" ]
 end
