--- a/Makefile	2017-11-05 02:26:11.000000000 +0000
+++ b/Makefile	2017-12-15 19:55:11.644799174 +0000
@@ -137,6 +137,19 @@
 	cp -r testcases/ $${DESTDIR}$(MISC_PATH)
 	cp -r dictionaries/ $${DESTDIR}$(MISC_PATH)
 
+uninstall:
+	rm -rf $${DESTDIR}$(MISC_PATH)/dictionaries
+	rm -rf $${DESTDIR}$(MISC_PATH)/testcases
+	rm -f $${DESTDIR}$(DOC_PATH)/README $${DESTDIR}$(DOC_PATH)/ChangeLog $${DESTDIR}$(DOC_PATH)/*.txt
+	rm -f $${DESTDIR}$(HELPER_PATH)/as $${DESTDIR}$(HELPER_PATH)/afl-as
+	set -e; for i in afl-g++ afl-clang afl-clang++; do rm -f $${DESTDIR}$(BIN_PATH)/$$i; done
+	rm -f $${DESTDIR}$(HELPER_PATH)/afl-llvm-rt*.o
+	rm -f $${DESTDIR}$(BIN_PATH)/afl-clang-fast $${DESTDIR}$(BIN_PATH)/afl-clang-fast++
+	rm -f $${DESTDIR}$(BIN_PATH)/afl-qemu-trace
+	set -e; for prog in $(PROGS) $(SH_PROGS); do rm -f $${DESTDIR}$(BIN_PATH)/$$prog; done
+	rm -f $${DESTDIR}$(BIN_PATH)/afl-plot.sh
+	
+
 publish: clean
 	test "`basename $$PWD`" = "afl" || exit 1
 	test -f ~/www/afl/releases/$(PROGNAME)-$(VERSION).tgz; if [ "$$?" = "0" ]; then echo; echo "Change program version in config.h, mmkay?"; echo; exit 1; fi

