diff --git b/support/META a/support/META
index 195448a..cc9881f 100644
--- b/support/META
+++ a/support/META
@@ -7,3 +7,15 @@ archive(byte) = "labltk.cma"
 archive(native) = "labltk.cmxa"
 linkopts = ""
 
+package "jpf" (
+description = "a 'file selector' and 'balloon help' support for labltk"
+requires = "unix,labltk"
+archive(byte) = "jpflib.cma"
+archive(native) = "jpflib.cmxa"
+)
+package "frx" (
+description = "Francois Rouaix's widget set library"
+requires = "unix,labltk"
+archive(byte) = "frxlib.cma"
+archive(native) = "frxlib.cmxa"
+)
