diff --git a/dune b/dune
index 04a65be..b43aaf7 100644
--- a/dune
+++ b/dune
@@ -18,4 +18,5 @@
                echo let summary = major, minor, patch, git_revision")
     )
   )
+  (mode fallback)
 )
diff --git a/ordma_version.ml b/ordma_version.ml
new file mode 100644
index 0000000..b97886f
--- /dev/null
+++ b/ordma_version.ml
@@ -0,0 +1,3 @@
+let git_revision = "tags/0.0.5-0-g71dad88"
+let major, minor, patch = 0,0,5
+let summary = major, minor, patch, git_revision
