diff --git a/_tags b/_tags
index 1a67ce9..cb802d7 100644
--- a/_tags
+++ b/_tags
@@ -6,7 +6,7 @@ true: bin_annot, safe_string, debug, warn(-58)
 <test> : include
 
 <src/*>: warn(@5@8@9@10@11@12@14@15@16@23@24@25@26@27@29@33@40)
-<src/*>: package(ppx_core), package(ppx_tools.metaquot)
+<src/*>: package(ppx_ast), package(ppx_core), package(ppx_tools.metaquot)
 
 <test/*>:  warn(@5@8@10@11@12@14@23@24@26@29@40)
 <test/*>:  use_ppx_fun
