diff --git a/lib/mustache.ml b/lib/mustache.ml
index f6ffcc7..2c702fa 100644
--- a/lib/mustache.ml
+++ b/lib/mustache.ml
@@ -5,6 +5,9 @@ module Str = Re_str
 module List = ListLabels
 module String = StringLabels
 
+let (|>) x f = f x
+let (@@) f x = f x
+
 exception Invalid_param of string with sexp
 exception Invalid_template of string with sexp
 
diff --git a/lib_test/mustache_cli.ml b/lib_test/mustache_cli.ml
index ff90ae9..49ee0ba 100644
--- a/lib_test/mustache_cli.ml
+++ b/lib_test/mustache_cli.ml
@@ -1,3 +1,5 @@
+let (|>) x f = f x
+let (@@) f x = f x
 
 let apply_mustache json_data template_data =
 	let env = Ezjsonm.from_string json_data
diff --git a/lib_test/test_mustache.ml b/lib_test/test_mustache.ml
index 2b49f5a..2c6bc1d 100644
--- a/lib_test/test_mustache.ml
+++ b/lib_test/test_mustache.ml
@@ -2,6 +2,9 @@ open OUnit2
 open Printf
 open Sexplib.Std
 
+let (|>) x f = f x
+let (@@) f x = f x
+
 module List = ListLabels
 module String = StringLabels
 
