diff --git a/Makefile.conf b/Makefile.conf
index 2be0c43..650cf23 100644
--- a/Makefile.conf
+++ b/Makefile.conf
@@ -49,8 +49,6 @@ NATDYNLINK ?= $(shell if [ -f `ocamlc -where`/dynlink.cmxa ]; then echo YES; els
 
 WITH_PPX_TOOLS ?= $(shell if [ -f `ocamlfind query ppx_tools 2> /dev/null`/ppx_tools.cma ]; then echo YES; else echo NO; fi)
 
-WITH_PPX_DERIVING ?= $(shell if [ -f `ocamlfind query ppx_deriving 2> /dev/null`/ppx_deriving.cma ]; then echo YES; else echo NO; fi)
-
 ##disabled for ocaml < 4.02.2+trunk
 ifeq "${WITH_PPX_TOOLS}" "YES"
 ifneq ($(shell ocamlc -version | grep -q -E "4.02.[01]"; echo $$?),0)
