diff -ru ctypes.0.3.4/src/ctypes/type_info_stubs.c ctypes.0.3.4_/src/ctypes/type_info_stubs.c
--- ctypes.0.3.4/src/ctypes/type_info_stubs.c	2014-12-23 01:11:05.000000000 +0100
+++ ctypes.0.3.4_/src/ctypes/type_info_stubs.c	2015-02-27 15:43:17.232556328 +0100
@@ -48,7 +48,7 @@
    case Uint8_t: b = ctypes_copy_uint8(*(uint8_t *)buf); break;
    case Uint16_t: b = ctypes_copy_uint16(*(uint16_t *)buf); break;
    case Uint32_t: b = ctypes_copy_uint32(*(uint32_t *)buf); break;
-   case Uint64_t: b = ctypes_copy_uint64(*(uint64 *)buf); break;
+   case Uint64_t: b = ctypes_copy_uint64(*(uint64_t *)buf); break;
    case Camlint: b = Val_int(*(intnat *)buf); break;
    case Nativeint: b = caml_copy_nativeint(*(intnat *)buf); break;
    case Float: b = caml_copy_double(*(float *)buf); break;
@@ -89,7 +89,7 @@
    case Uint8_t: *(uint8_t *)buf = Uint8_val(v); break;
    case Uint16_t: *(uint16_t *)buf = Uint16_val(v); break;
    case Uint32_t: *(uint32_t *)buf = Uint32_val(v); break;
-   case Uint64_t: *(uint64 *)buf = Uint64_val(v); break;
+   case Uint64_t: *(uint64_t *)buf = Uint64_val(v); break;
    case Camlint: *(intnat *)buf = Int_val(v); break;
    case Nativeint: *(intnat *)buf = Nativeint_val(v); break;
    case Float: *(float *)buf = Double_val(v); break;
