diff -C 2 -r -w camldbm-1.0.orig/Makefile camldbm-1.0/Makefile
*** camldbm-1.0.orig/Makefile	2011-11-22 16:56:49.000000000 +0100
--- camldbm-1.0/Makefile	2016-03-18 15:26:56.032475666 +0100
***************
*** 58,62 ****
  
  install::
! 	if test -f dllcamldbm.$(SO); then cp dllcamldbm.$(SO) $(STUBLIBDIR)/; fi 
  	cp libcamldbm.$(A) $(LIBDIR)/
  	cd $(LIBDIR) && ranlib libcamldbm.$(A)
--- 58,62 ----
  
  install::
! 	if test -f dllcamldbm.$(SO); then mkdir $(STUBLIBDIR) || echo Ok; cp dllcamldbm.$(SO) $(STUBLIBDIR)/; fi 
  	cp libcamldbm.$(A) $(LIBDIR)/
  	cd $(LIBDIR) && ranlib libcamldbm.$(A)
Only in camldbm-1.0: Makefile~
