diff -u -r -N cgi-0.8-orig/META cgi-0.8/META
--- cgi-0.8-orig/META	1970-01-01 01:00:00.000000000 +0100
+++ cgi-0.8/META	2013-02-18 13:29:47.000000000 +0100
@@ -0,0 +1,5 @@
+description = "Objective Caml library for writing CGIs"
+version = "0.8"
+requires = "str"
+archive(byte) = "cgi.cmo"
+archive(native) = "cgi.cmx"
diff -u -r -N cgi-0.8-orig/Makefile.in cgi-0.8/Makefile.in
--- cgi-0.8-orig/Makefile.in	2013-02-18 13:29:26.000000000 +0100
+++ cgi-0.8/Makefile.in	2013-02-18 13:30:40.000000000 +0100
@@ -3,8 +3,6 @@
 # Configuration part
 ###################################################################
 
-TARGETDIR = @OCAMLLIB@
-
 CAMLC   = @OCAMLC@ -g
 CAMLOPT = @OCAMLOPT@
 CAMLDEP = @OCAMLDEP@
@@ -28,20 +26,17 @@
 # Installation and export
 ###################################################################
 
-COMMFILES = $(MLI) $(CMI)
+COMMFILES = META $(MLI) $(CMI)
 BYTEFILES = cgi.cmo
 OPTFILES  = cgi.cmx cgi.o
 
 install: install-@OCAMLBEST@
 
-install-common:
-	cp $(COMMFILES) $(TARGETDIR)
-
-install-byte: install-common
-	cp $(BYTEFILES) $(TARGETDIR)
+install-byte:
+	ocamlfind install cgi $(COMMFILES) $(BYTEFILES)
 
-install-opt: install-common install-byte
-	cp $(OPTFILES) $(TARGETDIR)
+install-opt:
+	ocamlfind install cgi $(COMMFILES) $(BYTEFILES) $(OPTFILES)
 
 MAJORVN=0
 MINORVN=8
diff -u -r -N cgi-0.8-orig/cgi-0.8/META cgi-0.8/cgi-0.8/META
--- cgi-0.8-orig/cgi-0.8/META	1970-01-01 01:00:00.000000000 +0100
+++ cgi-0.8/cgi-0.8/META	2013-02-18 13:29:44.000000000 +0100
@@ -0,0 +1,5 @@
+description = "Objective Caml library for writing CGIs"
+version = "0.8"
+requires = "str"
+archive(byte) = "cgi.cmo"
+archive(native) = "cgi.cmx"
