diff -r -C 2 ocp-index.1.1.4.orig/opam ocp-index.1.1.4/opam
*** ocp-index.1.1.4.orig/opam	2016-05-12 11:00:19.040179931 +0200
--- ocp-index.1.1.4/opam	2016-05-12 11:00:58.084181600 +0200
***************
*** 14,18 ****
    "org:typerex"
  ]
! dev-repo: "https://github.com/OCamlPro/ocp-index.git#4.02"
  build: [
    ["./configure" "--prefix" prefix]
--- 14,18 ----
    "org:typerex"
  ]
! dev-repo: "https://github.com/OCamlPro/ocp-index.git#4.03"
  build: [
    ["./configure" "--prefix" prefix]
***************
*** 27,31 ****
  depopts: "lambda-term"
  conflicts: "lambda-term" {< "1.7"}
! available: [ocaml-version >= "4.02.0" & ocaml-version < "4.03"]
  messages: "For ocp-browser, please also install package lambda-term" {! lambda-term:installed}
  post-messages:
--- 27,31 ----
  depopts: "lambda-term"
  conflicts: "lambda-term" {< "1.7"}
! available: [ocaml-version >= "4.03"]
  messages: "For ocp-browser, please also install package lambda-term" {! lambda-term:installed}
  post-messages:
diff -r -C 2 ocp-index.1.1.4.orig/src/grepMain.ml ocp-index.1.1.4/src/grepMain.ml
*** ocp-index.1.1.4.orig/src/grepMain.ml	2016-05-12 11:00:19.040179931 +0200
--- ocp-index.1.1.4/src/grepMain.ml	2016-05-12 11:00:58.152181603 +0200
***************
*** 86,90 ****
            (try Filename.chop_extension f with Invalid_argument _ -> f)
        in
!       String.mapi (function 0 -> Char.uppercase | _ -> fun x -> x) s
      in
      let f (curpath, lookfor, last_scope, acc) scope tok pos =
--- 86,90 ----
            (try Filename.chop_extension f with Invalid_argument _ -> f)
        in
!       String.mapi (function 0 -> Char.uppercase_ascii | _ -> fun x -> x) s
      in
      let f (curpath, lookfor, last_scope, acc) scope tok pos =
diff -r -C 2 ocp-index.1.1.4.orig/src/indexBuild.ml ocp-index.1.1.4/src/indexBuild.ml
*** ocp-index.1.1.4.orig/src/indexBuild.ml	2016-05-12 11:00:19.040179931 +0200
--- ocp-index.1.1.4/src/indexBuild.ml	2016-05-12 11:00:58.152181603 +0200
***************
*** 222,225 ****
--- 222,226 ----
          Otyp_module (str, strl, List.map aux tylist)
      | Otyp_open -> Otyp_open
+     | Otyp_attribute (ty,attr) -> Otyp_attribute(aux ty,attr)
    in
    aux ty
***************
*** 234,238 ****
          otype_cstrs = List.map (fun (ty1,ty2) -> qual ty1, qual ty2)
                            out_type_decl.otype_cstrs }, rc)
!   | Osig_value (str, ty, str2) -> Osig_value (str, qual ty, str2)
    | Osig_typext (constr, es) ->
        Osig_typext ({ constr with
--- 235,239 ----
          otype_cstrs = List.map (fun (ty1,ty2) -> qual ty1, qual ty2)
                            out_type_decl.otype_cstrs }, rc)
!   | Osig_value v -> Osig_value { v with oval_type = qual v.oval_type }
    | Osig_typext (constr, es) ->
        Osig_typext ({ constr with
***************
*** 304,308 ****
    | _, PStr [{pstr_desc = Pstr_eval ({pexp_desc},_)}] ->
        (match pexp_desc with
!        | Pexp_constant (Const_string (s,_)) -> Some s
         | _ -> debug "Unexpected ocaml.doc docstring format"; None)
    | _ -> None
--- 305,309 ----
    | _, PStr [{pstr_desc = Pstr_eval ({pexp_desc},_)}] ->
        (match pexp_desc with
!        | Pexp_constant (Pconst_string (s,_)) -> Some s
         | _ -> debug "Unexpected ocaml.doc docstring format"; None)
    | _ -> None
***************
*** 323,326 ****
--- 324,328 ----
                  otype_type    = ty;
                  otype_private = Asttypes.Public;
+                 otype_immediate = false;
                  otype_cstrs   = []; }, Outcometree.Orec_not)
            in
***************
*** 345,354 ****
            let ty =
              let params = match cd_args with
!               | [] -> Outcometree.Otyp_sum []
!               | param::_ ->
                       Printtyp.tree_of_typexp false
!                        { Types. desc = Types.Ttuple cd_args;
                           level = param.Types.level;
                           id = param.Types.id }
              in
              Outcometree.Osig_type (Outcometree.{
--- 347,365 ----
            let ty =
              let params = match cd_args with
!               | Cstr_tuple [] -> Outcometree.Otyp_sum []
!               | Cstr_tuple (param::_ as l) ->
                       Printtyp.tree_of_typexp false
!                        { Types. desc = Types.Ttuple l;
                           level = param.Types.level;
                           id = param.Types.id }
+               | Cstr_record params ->
+                   Outcometree.Otyp_record (
+                     List.map
+                       (fun l ->
+                          (Ident.name l.Types.ld_id,
+                           l.ld_mutable = Mutable,
+                           Printtyp.tree_of_typexp false l.ld_type)
+                       )
+                       params)
              in
              Outcometree.Osig_type (Outcometree.{
***************
*** 357,360 ****
--- 368,372 ----
                  otype_type    = params;
                  otype_private = Asttypes.Public;
+                 otype_immediate = false;
                  otype_cstrs   = []; }, Outcometree.Orec_not)
            in
***************
*** 522,525 ****
--- 534,538 ----
                      otype_type    = ty;
                      otype_private = Asttypes.Public;
+                     otype_immediate = false;
                      otype_cstrs   = []; }, Outcometree.Orec_not)
                in
***************
*** 947,952 ****
        let i = String.rindex file '.' in
        let len = String.length file in
!       let modul = String.capitalize (String.sub file 0 i) in
!       let ext = String.lowercase (String.sub file (i+1) (len-i-1)) in
        modul, ext
      with Not_found -> file, ""
--- 960,965 ----
        let i = String.rindex file '.' in
        let len = String.length file in
!       let modul = String.capitalize_ascii (String.sub file 0 i) in
!       let ext = String.lowercase_ascii (String.sub file (i+1) (len-i-1)) in
        modul, ext
      with Not_found -> file, ""
diff -r -C 2 ocp-index.1.1.4.orig/src/indexOut.ml ocp-index.1.1.4/src/indexOut.ml
*** ocp-index.1.1.4.orig/src/indexOut.ml	2016-05-12 11:00:19.040179931 +0200
--- ocp-index.1.1.4/src/indexOut.ml	2016-05-12 11:00:58.152181603 +0200
***************
*** 182,187 ****
      | Osig_type ({ otype_type },_) ->
          tydecl fmt otype_type
!     | Osig_value (_,ty,_) ->
          !Oprint.out_type fmt ty
  
    let ty ?(colorise = no_color) fmt id =
--- 182,189 ----
      | Osig_type ({ otype_type },_) ->
          tydecl fmt otype_type
!     | Osig_value { oval_type = ty } ->
          !Oprint.out_type fmt ty
+     | Osig_ellipsis ->
+         Format.fprintf fmt "..."
  
    let ty ?(colorise = no_color) fmt id =
diff -r -C 2 ocp-index.1.1.4.orig/src/indexPredefined.ml ocp-index.1.1.4/src/indexPredefined.ml
*** ocp-index.1.1.4.orig/src/indexPredefined.ml	2016-05-12 11:00:19.040179931 +0200
--- ocp-index.1.1.4/src/indexPredefined.ml	2016-05-12 11:00:58.152181603 +0200
***************
*** 29,32 ****
--- 29,33 ----
          otype_type    = def;
          otype_private = Asttypes.Public;
+         otype_immediate = false;
          otype_cstrs   = [] }, Orec_not));
    loc_sig = Lazy.from_val Location.none;
***************
*** 46,49 ****
--- 47,51 ----
          otype_type    = (match params with [] -> Otyp_sum []
                                           | l  -> Otyp_tuple l);
+         otype_immediate = false;
          otype_private = Asttypes.Public;
          otype_cstrs   = [] }, Orec_not));
