--- ../Makefile.orig	2018-04-21 20:13:49.022952632 +0200
+++ ../Makefile	2018-04-21 20:14:10.727065199 +0200
@@ -15,8 +15,8 @@
 dllcamlpdf_stubs.* $(foreach x,$(PDFMODS),$x.mli) \
 $(foreach x,$(PDFMODS),$x.cmi) $(foreach x,$(PDFMODS),$x.cmx)
 
-OCAMLNCFLAGS = -g -annot -w -3
-OCAMLBCFLAGS = -g -annot -w -3
+OCAMLNCFLAGS = -g -annot -w -3 -unsafe-string
+OCAMLBCFLAGS = -g -annot -w -3 -unsafe-string
 OCAMLLDFLAGS = -g
 
 all : native-code-library byte-code-library htdoc
