From 50343916acec27b13c3e9101ca8bd84ecce98665 Mon Sep 17 00:00:00 2001
From: Marcello Seri <marcello.seri@citrix.com>
Date: Wed, 21 Jun 2017 15:21:08 +0100
Subject: [PATCH] Fix _oasis to use xen-gnt-unix

Signed-off-by: Marcello Seri <marcello.seri@citrix.com>
---
 _oasis | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/_oasis b/_oasis
index 0aa7f86..f8b96b1 100644
--- a/_oasis
+++ b/_oasis
@@ -12,7 +12,7 @@ Library rrd_transport
   Path:                 lib
   Findlibname:          rrd-transport
   Modules:              Rrd_json, Rrd_protocol, Rrd_rpc, Rrd_protocol_v1, Rrd_protocol_v2, Rrd_io, Rrd_reader, Rrd_writer
-  BuildDepends:         bigarray, bytes, cstruct, stringext, xcp.rrd, rrd, xen-gnt, xen-gnt.unix, crc, threads
+  BuildDepends:         bigarray, bytes, cstruct, stringext, xcp.rrd, rrd, xen-gnt, xen-gnt-unix, crc, threads
 
 Executable rrdreader
   CompiledObject:       best
-- 
2.11.0

