From 730dbb034559b3e549ec0b2973cd0400a3fa072f Mon Sep 17 00:00:00 2001
From: Markus Boehme <markubo@amazon.com>
Date: Tue, 18 Aug 2020 11:53:15 +0200
Subject: [PATCH] Makefile: Ensure target directory exists for `install-python`

Fixes: #196

Signed-off-by: Markus Boehme <markubo@amazon.com>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 48c42e3cb..e25174413 100644
--- a/Makefile
+++ b/Makefile
@@ -329,6 +329,7 @@ install-spgen : tools/spgen/source/spgen$(TOOLS_SUFFIX)
 
 .PHONY : install-python
 install-python:
+	$(MKDIR_P) $(DESTDIR)$(LIBDIR)/python/coccilib
 	$(INSTALL_DATA) python/coccilib/*.py \
 		$(DESTDIR)$(LIBDIR)/python/coccilib
 
