diff --git a/_tags b/_tags
index 69db3f4..5d39b96 100644
--- a/_tags
+++ b/_tags
@@ -10,7 +10,7 @@ true: keep_locs
 <{lib,app}/*>: package(unix dynlink cmdliner rresult fmt.tty ocamlgraph astring fpath bos logs logs.cli)
 
 # Force the runtime to be unix-independent.
-<runtime/*>: dontlink(unix str num threads)
+<runtime/*>: dontlink(unix str threads)
 <runtime/*>: package(cmdliner fmt)
 
 <tests/*>: package(oUnit cmdliner fmt.cli rresult astring bos logs logs.cli logs.fmt fmt.tty)
