diff --git a/Makefile b/Makefile
index b4c8c64..de27dde 100644
--- a/Makefile
+++ b/Makefile
@@ -20,6 +20,23 @@ byte: $(LOCAL_OCPBUILD)
 opt: $(LOCAL_OCPBUILD)
 	$(OCPBUILD) -asm
 
+OCAMLBUILD_FLAGS=\
+	-Is src/core,src/client,src/repositories,src/solver \
+	-use-ocamlfind -pkgs re.glob,ocamlgraph,cmdliner,cudf,dose \
+	-classic-display
+with-ocamlbuild: autogen
+	@for i in core repositories solver client; do\
+	  echo Compiling opam-$$i;\
+	  ls src/$$i/*.{ml,mll,mly} 2> /dev/null\
+	    | xargs basename\
+	    | awk -F. "{ print (toupper(substr(\$$1,0,1)) substr(\$$1,2)) }"\
+	    > src/$$i/opam-$$i.mllib;\
+	  ocamlbuild $(OCAMLBUILD_FLAGS) opam-$$i.cma opam-$$i.cmxa;\
+	  rm -f src/$$i/opam-$$i.mllib;\
+	done;\
+	ocamlbuild $(OCAMLBUILD_FLAGS) opamMain.native;\
+	ln -sf _build/src/client/opamMain.native opam
+
 $(LOCAL_OCPBUILD): ocp-build/ocp-build.boot ocp-build/win32_c.c
 	$(MAKE) -C ocp-build
 
@@ -39,6 +56,7 @@ clean:
 	rm -rf _obuild
 	rm -f *.annot src/*.annot
 	rm -f ocp-build.*
+	rm -f _build
 	$(MAKE) -C $(SRC_EXT) clean
 	$(MAKE) -C ocp-build clean
 
@@ -93,7 +111,7 @@ REPO_LIB   = opam-repositories
 SOLVER_LIB = opam-solver
 CLIENT_LIB = opam-client
 
-CORE_NOMLI = opamGlobals.ml
+CORE_NOMLI = opamGlobals.ml opamParser.ml opamLexer.ml opamLineLexer.ml
 CORE_MLI   = $(foreach i, $(shell find src/core -name "*.mli"), $(notdir $i))
 REPO_MLI   = $(foreach i, $(shell find src/repositories -name "*.mli"), $(notdir $i))
 SOLVER_MLI = $(foreach i, $(shell find src/solver -name "*.mli"), $(notdir $i))
@@ -113,9 +131,17 @@ FILES = $(CORE_FILES:%=_obuild/opam-core/%)\
 	$(SOLVER_FILES:%=_obuild/opam-solver/%)\
 	$(CLIENT_FILES:%=_obuild/opam-client/%)
 
+OCAMLBUILD_FILES =\
+	$(CORE_FILES:%=_build/src/core/%)\
+	$(REPO_FILES:%=_build/src/repositories/%)\
+	$(SOLVER_FILES:%=_build/src/solver/%)\
+	$(CLIENT_FILES:%=_build/src/client/%)
+
 .PHONY: libuninstall libinstall
 libinstall: META
 	ocamlfind install opam META $(FILES)
+libinstall-with-ocamlbuild: META
+	ocamlfind install opam META $(OCAMLBUILD_FILES)
 libuninstall:
 	ocamlfind remove opam
 
