Index: configure
===================================================================
--- configure	(revision 652)
+++ configure	(working copy)
@@ -594,9 +594,11 @@
 CAMLP4_LIB
 OCF_INCLUDE
 OCF_PATH
+YOJSON_INCLUDE
 YOJSON_PATH
 BINIOU_INCLUDE
 BINIOU_PATH
+EASY_FORMAT_INCLUDE
 EASY_FORMAT_PATH
 OCamlCairo2_INCLUDE
 OCamlCairo2_PATH
@@ -1322,8 +1324,8 @@
               Directory where to find the OCaml Cairo bindings if not in a
               standard location
   EASY_FORMAT_PATH
-              Directory where to find the Yojson library if not in a standard
-              location
+              Directory where to find the Easy-format library if not in a
+              standard location
   BINIOU_PATH Directory where to find the biniou library if not in a standard
               location
   YOJSON_PATH Directory where to find the Yojson library if not in a standard
@@ -3429,6 +3431,8 @@
 if test "$FOUND_LIB" != "no" ; then
    EASY_FORMAT_PATH=$LIB_PATH
 
+   EASY_FORMAT_INCLUDE=$LIB_INCLUDE
+
 fi
 
 # Look for biniou with ocamlfind
@@ -3771,6 +3775,8 @@
 if test "$FOUND_LIB" != "no" ; then
    YOJSON_PATH=$LIB_PATH
 
+   YOJSON_INCLUDE=$LIB_INCLUDE
+
 fi
 
 # Look for ocf with ocamlfind
