Index: config/configure.ac
===================================================================
--- config/configure.ac	(revision 656)
+++ config/configure.ac	(working copy)
@@ -186,11 +186,12 @@
 fi
 
 # Look for Easy-format with ocamlfind
-AC_ARG_VAR(EASY_FORMAT_PATH,[Directory where to find the Yojson library if not in a standard location])
+AC_ARG_VAR(EASY_FORMAT_PATH,[Directory where to find the Easy-format library if not in a standard location])
 AC_LIB_CHECKING(Easy-format,easy-format,easy_format,easy-format,easy-format,Easy_format,"",true,$EASY_FORMAT_PATH)
 
 if test "$FOUND_LIB" != "no" ; then
    AC_SUBST(EASY_FORMAT_PATH,$LIB_PATH)
+   AC_SUBST(EASY_FORMAT_INCLUDE,$LIB_INCLUDE)
 fi
 
 # Look for biniou with ocamlfind
@@ -207,6 +208,7 @@
 
 if test "$FOUND_LIB" != "no" ; then
    AC_SUBST(YOJSON_PATH,$LIB_PATH)
+   AC_SUBST(YOJSON_INCLUDE,$LIB_INCLUDE)
 fi
 
 # Look for ocf with ocamlfind
