Index: src/scripting/Makefile.in
===================================================================
--- src/scripting/Makefile.in	(revision 635)
+++ src/scripting/Makefile.in	(working copy)
@@ -26,12 +26,12 @@
 ###############################
 
 # Used libraries
-LIBS += dyp.cma str.cma ANSITerminal.cma bigarray.cma cairo2.cma @EASY_FORMAT_PATH@/easy_format.cmo biniou.cma @YOJSON_PATH@/yojson.cmo ocf.cma
+LIBS += dyp.cma str.cma ANSITerminal.cma bigarray.cma cairo2.cma easy_format.cma biniou.cma yojson.cma ocf.cma
 
 # The corresponding directories
 # (if not in the main ocaml lib directory,
 # ex. -I  +campl4
-LIBDIR += @DYPGEN_INCLUDE@ @ANSITerminal_INCLUDE@ @OCamlCairo2_INCLUDE@ @BINIOU_INCLUDE@ @OCF_INCLUDE@
+LIBDIR += @DYPGEN_INCLUDE@ @ANSITerminal_INCLUDE@ @OCamlCairo2_INCLUDE@ @EASY_FORMAT_INCLUDE@ @YOJSON_INCLUDE@ @BINIOU_INCLUDE@ @OCF_INCLUDE@
 
 # Directories to which the current source files depend on
 PREVIOUS_DIRS = ../utils ../datalog.prover ../logic ../acg-data ../grammars
