diff --git a/configure b/configure
index 49c02df..165e210 100755
--- a/configure
+++ b/configure
@@ -4227,7 +4227,7 @@ if test "$OCAMLBEST" = "opt"; then :
 else
   OCAMLBESTCC=$OCAMLC
 fi
-OCAML_CC="$($OCAMLBESTCC -config | fgrep native_c_compiler | sed -e "s/native_c_compiler: \(.*\) .*/\1/")"
+OCAML_CC="$($OCAMLBESTCC -config | grep "^native_c_compiler:" | sed -e "s/native_c_compiler: \(.*\) .*/\1/")"
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
@@ -5315,7 +5315,7 @@ done
   HAS_RPM=yes
 fi
 
-OCAML_SYSTEM="$($OCAMLBESTCC -config | fgrep system | sed -e "s/system: \(.*\)/\1/")"
+OCAML_SYSTEM="$($OCAMLBESTCC -config | grep "^system:" | sed -e "s/system: \(.*\)/\1/")"
 
 
 if test "${OCAML_OS_TYPE}" = "Win32"; then :
