diff --git a/cmake/modules/AddOCaml.cmake b/cmake/modules/AddOCaml.cmake
index 1b805c0710a..f5255f145b0 100644
--- a/cmake/modules/AddOCaml.cmake
+++ b/cmake/modules/AddOCaml.cmake
@@ -70,6 +70,7 @@ function(add_ocaml_library name)
   foreach( llvm_lib ${llvm_libs} )
     list(APPEND ocaml_flags "-l${llvm_lib}" )
   endforeach()
+  list(APPEND ocaml_flags "-ccopt" "-L%%LIBDIR%%" )
 
   get_property(system_libs TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
   foreach(system_lib ${system_libs})
