diff -u -r ocamlgraph-1.8.3/Makefile.in ocamlgraph-new/Makefile.in
--- ocamlgraph-1.8.3/Makefile.in	2013-04-17 20:30:17.000000000 +0200
+++ ocamlgraph-new/Makefile.in	2013-08-27 15:01:54.395662054 +0200
@@ -373,7 +373,7 @@
 		$(SRCDIR)/*.mli $(VIEWER_DIR)/*.mli $(DGRAPH_DIR)/*.mli \
 		graph$(LIBEXT) graph.cmx graph.cmo graph.cmi $(CMA) $(CMXA) \
 		$(VIEWER_CMXLIB) $(VIEWER_CMOLIB) $(DGRAPH_CMXLIB) \
-		$(DGRAPH_CMOLIB)
+		$(DGRAPH_CMOLIB) $(DGRAPH_CMILIB)
 else
 	$(OCAMLFIND) install ocamlgraph META \
 		$(SRCDIR)/*.mli $(VIEWER_DIR)/*.mli $(DGRAPH_DIR)/*.mli \
Seulement dans ocamlgraph-new: Makefile.in~
